---
title: "RCC_LSCC_TLS_paper"
author: "Karina"
date: "2023-04-13"
output: html_document
---

```{r packages}


library(DESeq2)
library(EnhancedVolcano)
library(pals)
library("ggsci")
library(ggpubr)
library(dplyr)
library(data.table)
library(ggplot2)
library(dbscan)
library(igraph)

library(tidyverse)
library(rstatix)
library(RColorBrewer)

#library(Homo.sapiens.hg38)
library(DOSE)
library(clusterProfiler)
library(fgsea)

(mydir <- getwd())
#sessionInfo()
#packageVersion("pheatmap")
```

##Load data
```{r  }
pD0 <- read.csv("SegmentProperties.csv")
pD0 <- pD0 %>% mutate(Tissue_Subtype = dplyr::recode(Tissue_Subtype,
  "TLSEp" = "TLSep"
))

table(pD0$QC)
pD0[pD0$QC%like%"NO_READS",]

m.raw <- read.csv("TargetCountMatrix.csv",row.names=1)
colnames(m.raw) <- gsub("\\.\\.\\.","-",colnames(m.raw))
Meta_data_ordered = data.frame(AOI = colnames(m.raw))

fD <- read.csv("TargetProperties.csv",row.names=1)

pD0$AOI <- Meta_data_ordered$AOI
pD0$SegmentDisplayName <- gsub("\\|","-",pD0$SegmentDisplayName)
pD0$SegmentDisplayName <- gsub(" ","",pD0$SegmentDisplayName)
pD0$SegmentDisplayName <- gsub("GeometricSegment","Geometric.Segment",pD0$SegmentDisplayName)

#Rename LOQ to make my life easier
colnames(pD0)[grepl("LOQ",colnames(pD0))] <- "LOQ"
rownames(pD0) <- pD0$SegmentDisplayName

#add dataset filtering parameters
pD0$Immune <- grepl("TLS$|LN$",pD0$Tissue_type)
pD0$EPITH <- grepl("Alveoles$|TLSep$|Kidney$|LSCC$|RCC$",pD0$Tissue_Subtype)
pD0$TLSep <- grepl("TLSep$",pD0$Tissue_Subtype)

RNA <- read.csv("FFPE_RNAextraction_samples.csv")

```

# Compute variables
```{r  }
m.end <- m.raw[fD$CodeClass=="Endogenous",]
m.negs <- m.raw[fD$CodeClass!="Endogenous",]
neg.count <- t(m.raw)[,fD$CodeClass!="Endogenous"]
m.snr <- t(t(m.end)/neg.count)
pD0$totalsum <- colSums(m.raw)
pD0$totalsumwoneg <- colSums(m.end)
pD0$dt <- colSums(m.snr>2)
pD0$negcount <- neg.count
pD0$Sample <- rownames(pD0)
pD0$q3 <- apply(m.end,2,function(AOI) quantile(AOI,.75))

##check the difference between gMean of Negative probes and LOQ
pD0$LOQvsNegCount <- pD0$LOQ/pD0$negcount

## Remove the sample with 10x higher background than the rest of the study
#pD <- pD[pD$AOI != "RCC1_20210505...071...Immune",]
rms <- which(neg.count>750)
m.end <- m.end[,-rms]
m.raw <- m.raw[,-rms]
m.snr <- m.snr[,-rms]
pD0 <- pD0[-rms,]

###Sample-specific LOQ
#Compute a matrix where entries are 1/0 depending on whether they are above sample-specific LOQ and add info to pD
m.abLOQ <- t(apply(m.end, 1, function(ROW) ROW > pD0$LOQ))
pD0$aboveLOQ <- colSums(m.abLOQ)
pD0$aboveLOQprcnt <- round(pD0$aboveLOQ/nrow(m.end) * 100,1)

###SNR based threshold
pD0$aboveSNRpercent <- round(pD0$dt/nrow(m.end) * 100,1)


###Add RNA quality data

setdiff(pD0$Patient_ID, RNA$Patient.ID)
mm <- match(pD0$Patient_ID, RNA$Patient.ID)
pD0$DV200percent <- RNA$X..of.Total[mm]


```

## Assess sample quality
```{r data}
fp <- arrange(pD0,dt)
fp$Sample <- factor(fp$Sample,levels=fp$Sample)

colnames(fp)
```

```{r,correlation data}
Cordata <- fp[,c(7:8,37:39,41:43,45:50)]
res <- rcorr(as.matrix(Cordata), type="spearman")
res$r
Cordata <- fp[,c(23:26,28,36,7:8,37:39,41:43,45:50)]
colnames(Cordata)
```

```{r,correlation plots}
ggplot(Cordata,aes(x = AOISurfaceArea, y = aboveLOQprcnt)) + 
  geom_point(aes(shape = Tumor_type, color=Tissue_type)) +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~EPITH)+
  geom_smooth(method=lm, color="black") + 
  #scale_y_continuous(trans = "log10") +
  scale_x_continuous(trans = "log10") +
  geom_hline(yintercept = 3)+
  labs(title="",
       x= "Segment_size",
      y = "% genes > LOQ") +
  theme_light() + 
    theme(text=element_text(size=12),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          legend.title = element_text(size=8),
          legend.text = element_text(size=8), 
          axis.text.x = element_text(angle = 45, hjust = 1),
          strip.text = element_text(colour='black')) 
ggsave(filename = "Correl_aboveLOQ_AOIsizeFacetedEPITH.pdf", width = 6, height = 4)

ggplot(Cordata,aes(x = DV200percent, y = aboveLOQprcnt)) + 
  geom_point(aes(shape = Tumor_type, color=Tissue_type)) +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~EPITH)+
  geom_smooth(method=lm, color="black") + 
  #scale_y_continuous(trans = "log10") +
  scale_x_continuous(trans = "log10") +
  geom_hline(yintercept = 3)+
  labs(title="",
       x= "DV200%",
      y = "% genes > LOQ") +
  theme_light() + 
    theme(text=element_text(size=12),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          legend.title = element_text(size=8),
          legend.text = element_text(size=8), 
          strip.text = element_text(colour='black')) 
ggsave(filename = "Correl_aboveLOQ_DV200FacetedEPITH.pdf", width = 6, height = 4)


ggplot(Cordata,aes(x = DV200percent, y = totalsumwoneg)) + 
  geom_point(aes(shape = Tumor_type, color=Patient_ID)) +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~EPITH)+
  geom_smooth(method=lm, color="black") + 
  scale_y_continuous(trans = "log10") +
  #scale_x_continuous(trans = "log10") +
  labs(title="",
       x= "DV200%",
      y = "Total reads") +
  theme_light() + 
    theme(text=element_text(size=12),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          legend.title = element_text(size=8),
          legend.text = element_text(size=8), 
          strip.text = element_text(colour='black')) 
ggsave(filename = "Correl_totalreads_DV200FacetedEPITH.pdf", width = 6, height = 4)


ggplot(Cordata,aes(x = DV200percent, y = aboveLOQ)) + 
  geom_point(aes(shape = Tumor_type, color=Patient_ID)) +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~EPITH)+
  geom_smooth(method=lm, color="black") + 
  scale_y_continuous(trans = "log10") +
  #scale_x_continuous(trans = "log10") +
  labs(title="",
       x= "DV200%",
      y = "UMIs > LOQ") +
  theme_light() + 
    theme(text=element_text(size=12),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          legend.title = element_text(size=8),
          legend.text = element_text(size=8), 
          strip.text = element_text(colour='black')) 
ggsave(filename = "Correl_aboveLOQumis_DV200FacetedEPITH.pdf", width = 6, height = 4)



ggplot(Cordata,aes(x = DV200percent, y = negcount)) + 
  geom_point(aes(shape = Tumor_type, color=Patient_ID)) +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~EPITH)+
  geom_smooth(method=lm, color="black") + 
  scale_y_continuous(trans = "log10") +
  #scale_x_continuous(trans = "log10") +
  labs(title="",
       x= "DV200%",
      y = "Negative counts") +
  theme_light() + 
    theme(text=element_text(size=12),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          legend.title = element_text(size=8),
          legend.text = element_text(size=8), 
          strip.text = element_text(colour='black')) 
ggsave(filename = "Correl_Backgr_DV200FacetedEPITH.pdf", width = 6, height = 4)

```

```{r, plot threshold for bad LOQ AOIs}
ggplot(fp, aes(x=aboveLOQprcnt)) +
    geom_histogram(bins = 120) +
    #scale_x_log10() +
    theme_light()+
  facet_wrap(~Tissue_type,scales="free_x") +
  geom_vline(xintercept=3)+
    xlab("% genes above LOQ")+
    ylab("samples")
ggsave(filename = "PercentgenesaboveLOQ_tissuefaceted.pdf", width = 5, height = 4)

#strict threshold is 5%, forgiving threshold is 2.5%

ggplot(fp, aes(x=aboveLOQprcnt)) +
    geom_histogram(bins = 120) +
    #scale_x_log10() +
    theme_light()+
  facet_grid(Patient_ID~Tissue_type,scales="free_x") +
  geom_vline(xintercept=3)+
    xlab("% genes above LOQ")+
    ylab("samples")
ggsave(filename = "PercentgenesaboveLOQ_facetedPatient.pdf", width = 10, height = 6)

```

```{r, Define exclusion list}

#Using  LOQ filtering
tb.rmLOC <- unique(as.character(fp$Sample[fp$aboveLOQprcnt < 3]))
```

```{r check which AOIs get removed by which filtering}
pD1 <- pD0[pD0$Sample %in% tb.rm,]

df <-table(pD1$Tissue_Subtype, pD1$Patient_ID) %>% melt()
ggplot(data=df, aes(x=factor(Var2), y=value, fill=Var1)) +
  geom_bar(stat="identity") + 
  scale_fill_manual(values=cols25(25)) +
  labs(x = "Patient ID",
       y = "# of AOIs",
       fill = "Tissue")
#ggsave(filename = ".pdf", width = 8, height = 6)

pD1 <- pD0[pD0$Sample %in% tb.rmLOC,]

df <-table(pD1$Tissue_Subtype, pD1$Patient_ID) %>% melt()
ggplot(data=df, aes(x=factor(Var2), y=value, fill=Var1)) +
  geom_bar(stat="identity") + 
  scale_fill_manual(values=cols25(25)) +
  labs(x = "Patient ID",
       y = "# of AOIs",
       fill = "Tissue")

pD1 <- pD0[pD0$Sample %in% poorQ3,]

df <-table(pD1$Tissue_Subtype, pD1$Patient_ID) %>% melt()
ggplot(data=df, aes(x=factor(Var2), y=value, fill=Var1)) +
  geom_bar(stat="identity") + 
  scale_fill_manual(values=cols25(25)) +
  labs(x = "Patient ID",
       y = "# of AOIs",
       fill = "Tissue")

pD1 <- pD0[pD0$Sample %in% poorSNR,]

df <-table(pD1$Tissue_Subtype, pD1$Patient_ID) %>% melt()
ggplot(data=df, aes(x=factor(Var2), y=value, fill=Var1)) +
  geom_bar(stat="identity") + 
  scale_fill_manual(values=cols25(25)) +
  labs(x = "Patient ID",
       y = "# of AOIs",
       fill = "Tissue")

```

##Define final datasets
```{r filter AOIs}
#All filters
#pD <- pD0[!pD0$Sample %in% tb.rm,]

#Only LOQ filter
pD <- pD0[!pD0$Sample %in% tb.rmLOC,]

pD$Sample <- rownames(pD)
m.end <- m.end[,pD$Sample]
m.raw <- m.raw[,pD$Sample]
m.snr <- m.snr[,pD$Sample]
m.abLOQ <- m.abLOQ[,pD$Sample]

pD.im <- pD[pD$Immune,]
table(pD.im$Tissue_type)

discard <- c("LNmet", "TLSep")
pD.TLStum <- pD[!pD$Tissue_type%in%discard,]
pD.TLStumWithBadLOQ <- pD0[!pD0$Tissue_type%in%discard,]
table(pD.TLStum$Tissue_type)

pD.lscc <- pD.TLStum[pD.TLStum$Tumor_type == "LSCC",]
pD.lscc <- pD.lscc[pD.lscc$Tissue_type != "LN",]
table(pD.lscc$Tissue_type)

pD.Lsscim <- pD.im[pD.im$Tumor_type == "LSCC",]
pD.lscctls <- pD.Lsscim[pD.Lsscim$Tissue_type != "LN",]
pD.lscctls <- pD.lscctls[!pD.lscctls$Tissue_Subtype%like%"TLS_",]

pD.tls <- pD.im[pD.im$Tissue_type!="LN",]
pD.tls <- pD.tls[!pD.tls$Tissue_Subtype%like%"TLS_",]

pD.rcc <- pD.TLStum[pD.TLStum$Tumor_type == "RCC",]
table(pD.rcc$Tissue_type)

pD.rcctls <- pD.rcc[pD.rcc$Tissue_type%like%"TLS",]

select <- c("Alv", "Kid", "Tum")
pD.KiAl <- pD[pD$Tissue_type%in%select,]

pD.ep <- pD[pD$EPITH,]
pD.ep <- pD.ep[pD.ep$Tumor_type == "LSCC",]
pD.ep <- pD.ep[pD.ep$Tissue_type != "Tum",]

table(pD.TLStum$Tissue_type)
table(pD.lscctls$Tissue_Subtype, pD.lscctls$Tumor_type)
table(pD.rcctls$Tissue_Subtype, pD.rcctls$Tumor_type)

```

```{r filter Probes to make the final data set using LOQ threshold}
#TLS AOIs
numberofsamples <- length(unique(rownames(pD.tls)))
threshold <- numberofsamples*0.03/100
sums.tls <- rowSums(m.abLOQ[,rownames(pD.tls)])
prcnt.tls <- sums.tls/nrow(pD.tls)
keep.tls <- prcnt.tls>=threshold
m.tls.end <- m.end[keep.tls,rownames(pD.tls)]
Lib_sizeIm <- colSums(m.tls.end)

#TLS AOIs only in RCC
numberofsamples <- length(unique(rownames(pD.rcctls)))
threshold <- numberofsamples*0.05/100
sums.rcctls <- rowSums(m.abLOQ[,rownames(pD.rcctls)])
prcnt.tls <- sums.rcctls/nrow(pD.rcctls)
keep.tls <- prcnt.tls>=threshold
m.rcctls.end <- m.end[keep.tls,rownames(pD.rcctls)]
Lib_sizeIm <- colSums(m.rcctls.end)

#TLS AOIs only in LSCC
numberofsamples <- length(unique(rownames(pD.lscctls)))
threshold <- numberofsamples*0.05/100
sums.lcctls <- rowSums(m.abLOQ[,rownames(pD.lscctls)])
prcnt.tls <- sums.rcctls/nrow(pD.lscctls)
keep.tls <- prcnt.tls>=threshold
m.lscctls.end <- m.end[keep.tls,rownames(pD.lscctls)]
Lib_sizeIm <- colSums(m.lscctls.end)

#TLS+TUM+LN+ALV+KID AOIs
numberofsamples <- length(unique(rownames(pD.TLStum)))
threshold <- 14/numberofsamples
sums.tls <- rowSums(m.abLOQ[,rownames(pD.TLStum)])
prcnt.tls <- sums.tls/nrow(pD.TLStum)
keep.tls <- prcnt.tls>=threshold
m.tlstum.end <- m.end[keep.tls,rownames(pD.TLStum)]
Lib_sizeIm <- colSums(m.tlstum.end)

#TLS+LN AOIs
numberofsamples <- length(unique(rownames(pD.im)))
threshold <- numberofsamples*0.02/100
sums.tlsLN <- rowSums(m.abLOQ[,rownames(pD.im)])
prcnt.tlsLN <- sums.tlsLN/nrow(pD.im)
keep.tlsLN <- prcnt.tlsLN>=threshold
m.tlsLN.end <- m.end[keep.tlsLN,rownames(pD.im)]
Lib_sizeIm <- colSums(m.tlsLN.end)

#Kidney+Alveoles AOIs
numberofsamples <- length(unique(rownames(pD.KiAl)))
threshold <- numberofsamples*0.02/100
sums.parenchyme <- rowSums(m.abLOQ[,rownames(pD.KiAl)])
prcnt.parenchyme <- sums.parenchyme/nrow(pD.KiAl)
keep.parenchyme <- prcnt.parenchyme>=threshold
m.parenchyme.end <- m.end[keep.parenchyme,rownames(pD.KiAl)]
Lib_sizeparenchyme <- colSums(m.parenchyme.end)


#LSCC TLS+TUM+ALV AOIs
numberofsamples <- length(unique(rownames(pD.lscc)))
threshold <- 14/numberofsamples
sums.tls <- rowSums(m.abLOQ[,rownames(pD.lscc)])
prcnt.tls <- sums.tls/nrow(pD.lscc)
keep.tls <- prcnt.tls>=threshold
m.lscc.end <- m.end[keep.tls,rownames(pD.lscc)]
Lib_sizeIm <- colSums(m.lscc.end)

#RCC TLS+TUM+KID AOIs
numberofsamples <- length(unique(rownames(pD.rcc)))
threshold <- 14/numberofsamples
sums.tls <- rowSums(m.abLOQ[,rownames(pD.rcc)])
prcnt.tls <- sums.tls/nrow(pD.rcc)
keep.tls <- prcnt.tls>=threshold
m.rcc.end <- m.end[keep.tls,rownames(pD.rcc)]
Lib_sizeIm <- colSums(m.rcc.end)


#EPITHELIAL analysis
numberofsamples <- length(unique(rownames(pD.ep)))
threshold <- 4/numberofsamples
sums <- rowSums(m.abLOQ[,rownames(pD.ep)])
prcnt <- sums/nrow(pD.ep)
keep <- prcnt>=threshold
m.ep.end <- m.end[keep,rownames(pD.ep)]
Lib_sizeEP <- colSums(m.ep.end)


```


#Sample overview
```{r AOI numbers}
table(pD.TLStum$Tissue_type, pD.TLStum$Tumor_type)
table(pD.TLStum$Tumor_type)
table(pD.TLStumWithBadLOQ$Tissue_type, pD.TLStumWithBadLOQ$Tumor_type)
table(pD.TLStumWithBadLOQ$Tumor_type)

pD.TLStumWithBadLOQ <- pD.TLStumWithBadLOQ %>% 
  mutate(Tissue_type2 = recode(Tissue_type,
                               "Tum" = "Tumor",
                               "Kid" = "Parenchyma",
                               "Alv" = "Parenchyma")) %>%
  mutate(Tissue_type2 = factor(Tissue_type2, levels = c("Parenchyma", "Tumor", "TLS", "LN")))
  
ggplot(pD.TLStumWithBadLOQ, aes(x=aboveLOQprcnt)) +
    geom_histogram(bins = 100) +
    #scale_x_log10() +
    theme_light()+
  facet_wrap(~Tissue_type2,scales="free_x", nrow = 1) +
  geom_vline(xintercept=3, color = "Red")+
    xlab("% genes above LOQ")+
    ylab("samples")
ggsave(filename = "PercentgenesaboveLOQ_dataTLSTUM.pdf", width = 6, height = 3)
```

```{r good AOI barplot}
 pp_prefix <- "Barplot"
var <- "TLSTUMdata"
 plot = "goodAOI"
 
df <- pD.TLStum %>% 
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS",
  "LN_B" = "LN",
  "LN_GC" = "LN",
  "LN_T" = "LN"
  ))

df <- df %>% 
  group_by(Tumor_type, Patient_ID, Tissue_Subtype2) %>% 
  dplyr::summarize("number_of_AOIs" = n()) %>% 
  mutate(Tissue_Subtype2 = factor(Tissue_Subtype2, levels = c("E_TLS", "PFL_TLS","SFL_TLS","Tcell_TLS","LN" ,"Alveoles", "Kidney", "LSCC", "RCC"))) %>% 
  mutate(Tissue_Subtypecolor = factor(Tissue_Subtype2, levels = c("E_TLS", "SFL_TLS", "PFL_TLS", "Tcell_TLS","LN", "Alveoles", "Kidney", "LSCC", "RCC")))

ggplot(data=df, aes(x=factor(Patient_ID), y=number_of_AOIs, fill=Tissue_Subtypecolor)) +
  geom_bar(stat="identity") + 
  facet_wrap(~factor(Tumor_type), scales = "free")+
  scale_fill_manual(values=cols25(25)) +
  labs(x = "Patient ID",
       y = "# of AOIs",
       fill = "Tissue")+
theme_light() + 
    theme(text=element_text(size=22),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          axis.text.x = element_text(angle = 45, hjust = 1),
          #legend.position = "none"
          strip.text = element_text(colour='black'))
  
fn <- sprintf("%s_%s_%s.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 8, height = 6)

```

```{r calculate fractions of AOI types}
table(pD.TLStum$Tissue_type, pD.TLStum$Tumor_type)
data <- pD.TLStum %>% 
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS",
  "LN_B" = "LN",
  "LN_GC" = "LN",
  "LN_T" = "LN"
  ))

skip <- c("Tum", "LN", "Alv", "Kid")
d <- data[!data$Tissue_type%in%skip,]

totalAOIs <- d %>% 
    group_by(Tumor_type,Patient_ID, Patient_Category) %>%
    dplyr::summarize("number of AOIs" = n())

TcellAOIs <- d %>% filter(Tissue_Subtype2 == "Tcell_TLS") %>%
    group_by(Tumor_type,Patient_ID, Patient_Category) %>%
    dplyr::summarize("number of Tagg" = n())

eTLSAOIs <- d %>% filter(Tissue_Subtype2 == "E_TLS") %>%
    group_by(Tumor_type,Patient_ID, Patient_Category) %>%
    dplyr::summarize("number of E" = n())

pflTLSAOIs <- d %>% filter(Tissue_Subtype2 == "PFL_TLS") %>%
    group_by(Tumor_type,Patient_ID, Patient_Category) %>%
    dplyr::summarize("number of PFL" = n())

sflTLSAOIs <- d %>% filter(Tissue_Subtype2 == "SFL_TLS") %>%
    group_by(Tumor_type,Patient_ID, Patient_Category) %>%
    dplyr::summarize("number of SFL" = n())

AOIs <- merge(totalAOIs, TcellAOIs, by = "Patient_ID", all = TRUE)
AOIs <- merge(AOIs, eTLSAOIs, by = "Patient_ID", all = TRUE)
AOIs <- merge(AOIs, pflTLSAOIs, by = "Patient_ID", all = TRUE)
AOIs <- merge(AOIs, sflTLSAOIs, by = "Patient_ID", all = TRUE)
colnames(AOIs)

AOIcounts <- AOIs[,c(1:4, 7, 10, 13, 16)]
AOIcounts <- AOIcounts %>% mutate_all(~replace(., is.na(.), 0))
AOIfrac <- AOIcounts[,5:8]/AOIcounts[,4]
AOIstats <- cbind(AOIcounts[,1:3],AOIfrac)
colnames(AOIstats) <- c("Patient_ID", "Tumor_type","Survival", "Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS")

longAOI <- reshape2::melt(AOIstats,id = c("Patient_ID","Tumor_type", "Survival"))
longAOI <- longAOI %>% mutate(Tissue_Subtypecolor = factor(variable, levels = c("E_TLS", "SFL_TLS", "PFL_TLS", "Tcell_TLS", "Tumor")))
```

```{r long graphs}

 pp_prefix <- "Dotplot_mIF"
var <- "TLSTUMdata"
 plot = "AOIfractions"
 
ggplot(data = longAOI, aes(x = Tumor_type, y = value)) +
    geom_boxplot(alpha = 1, width = 0.4, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, size=4, pch=21, aes(fill=Tumor_type), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
  #scale_color_aaas() +
   scale_fill_manual(values=cols25(25)) +
    facet_wrap(~variable, nrow = 1) +
    stat_compare_means(comparisons=list(c("LSCC", "RCC")), size = 6) +
  labs(x = "Tumor type",
       y= "Fraction", #expression(paste("GC/cm"^2)),
      title = "Proportions of lymphocytic clusters"
      ) +
  #theme_pubr(border = TRUE) +
    theme_light() + 
    theme(text=element_text(size=22),
          plot.title = element_text(hjust = 0.5),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          #axis.text.x = element_text(angle = 45, hjust = 1),
          legend.position = "none",
          strip.text = element_text(colour='black'))

fn <- sprintf("%s_%s_%s.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 8, height = 6)



```

```{r unused}
ggplot(data = longAOI, aes(x = Tumor_type, y = value)) +
    geom_boxplot(alpha = 1, width = 0.4, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, width = 0.05, size=3, pch=21, aes(fill=Survival), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
  #scale_color_aaas() +
  scale_fill_manual(values=cols25(25)) +
    facet_wrap(~variable, nrow = 1) +
    stat_compare_means(comparisons=list(c("LSCC", "RCC")), size = 5) +
  labs(x = "Tumor type",
       y= "Fraction", #expression(paste("GC/cm"^2)),
      title = "Proportions of lymphocytic clusters"
      ) +
  #theme_pubr(border = TRUE) +
    theme_light()+
  theme(text=element_text(size=16),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          #strip.text = element_text(color='grey'),
        #axis.text.x = element_text(angle = 45, hjust = 1),
         # legend.title = element_text(size=12),
         # legend.text = element_text(size=12), 
          plot.title = element_text(size=16, hjust = 0.5))
 # theme(legend.position="none")

ggsave(filename = paste("mIF_AOIfractions", "inTumortypes_byTLStype_LOQ3filterTsurv.pdf", sep="_"), width = 7, height = 4)

ggplot(data = longAOI, aes(x = variable, y = value)) +
    geom_boxplot(alpha = 1, width = 0.4, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, width = 0.05, size=3, pch=21, aes(fill=Tumor_type), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
  #scale_color_aaas() +
   scale_fill_manual(values=cols25(25)) +
    facet_wrap(~Tumor_type , nrow = 1) +
  stat_compare_means(size = 5, aes(label=..p.adj..)) +
    #stat_compare_means(comparisons=list(c("PFL_TLS", "SFL_TLS"), c("SFL_TLS", "Tcell_agg")), size = 5) +
  labs(x = "Tumor type",
       y= "Fraction", #expression(paste("GC/cm"^2)),
      title = "Proportions of lymphocytic clusters"
      ) +
  #theme_pubr(border = TRUE) +
    theme_light()+
  theme(text=element_text(size=16),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          #strip.text = element_text(color='grey'),
        axis.text.x = element_text(angle = 45, hjust = 1),
         # legend.title = element_text(size=12),
         # legend.text = element_text(size=12), 
          plot.title = element_text(size=16, hjust = 0.5))+
  theme(legend.position="none")

ggsave(filename = paste("mIF_AOIfractions", "inTLStypes_byTumor_LOQ3filterT.pdf", sep="_"), width = 6, height = 4)
```


###Differential gene expression analysis

## all TLS 
# versus Lung
```{r load data}
cts <- as.matrix(m.lscc.end)
Meta_data_reorganised <- pD.lscc %>% mutate(Tissue_type = factor(Tissue_type))
unique(Meta_data_reorganised$Tissue_type)

Meta_data_reorganised <- Meta_data_reorganised %>% mutate(Tissue_type2 = dplyr::recode(Tissue_type,
  "Tum" = "Lung",
  "Alv" = "Lung"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_type2)

dim(cts)

table(dds$Tissue_type2)
```

```{r DESeq2 normalisation total TLS}
## set reference group for comparisons
dds$Tissue_type2 <- relevel(dds$Tissue_type2, ref = "TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results total TLS}

resLFC <- lfcShrink(dds, coef="Tissue_type2_Lung_vs_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_allTLSvsLung_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccALLTLSvsLungDEGsdata <- read.csv("DESeq2_lscc_allTLSvsLung_LOQ3.csv")
lsccALLTLSvsLungDEGsdata[lsccALLTLSvsLungDEGsdata$X%like%"IGH",]

lsccALLTLSvsLungDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.01]))



```

```{r Volcano total TLS vs lung}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_allTLSvsLung_LOQ3.csv")

Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -0.585 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 0.585 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Lung'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'         

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
lccALLTLSagainstLUNG <- subset(Dat, expression =="Down")
lsccLUNGaganistALLTLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- sort(as.vector(Df$X))

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 80),
    xlim = c(-3,4),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_TLSvsLUNG_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 8, height = 8)

```

```{r extract transformed values}
vsd <- vst(dds, blind=FALSE)
#rld <- rlog(dds, blind=FALSE)

dim(assay(vsd))
DESeq2normexprALL <- t(assay(vsd))
#DESeq2normexpr <- as.data.frame(DESeq2normexprALL[,differentialLSCC_allAGG_DEGs])

colnames(Meta_data_reorganised)
data <- Meta_data_reorganised[,c(1:4, 23:30, 35:38, 40, 44,47,48,50)]
data <- data %>%
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "LSCC" = "Lung",
  "Alveoles" = "Lung",
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  )) %>%
  mutate(Tissue_Subtypefacet = factor(Tissue_Subtype2, levels = c("Lung", "Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS"))) %>% 
  mutate(Tissue_Subtypecolor = factor(Tissue_Subtype2, levels = c("E_TLS", "SFL_TLS", "PFL_TLS", "Tcell_TLS", "Lung"))) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype,
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS")
)
#data <- cbind(data, DESeq2normexpr)

colnames(data)
unique(data$Patient_Category)
endcol <-ncol(data)



```

```{r, heatmap all TLS vs lung, fig.width=6, fig.height=4}

selectedgenes <- sort(c(significant))
exclude <- c()
selectedgenes <- selectedgenes[!selectedgenes%in%exclude]

hm <- assay(vsd[selectedgenes,])

#row annotation
geneset <- data.frame(genes = rownames(hm), Gene_set = "Other")
rownames(geneset) <- rownames(hm)

#recode gene set
geneset[geneset$gene%in%lccTCELLagainstLUNG$X,]$Gene_set <- "Tcell_TLS"
geneset[geneset$gene%in%lccETLSagainstLUNG$X,]$Gene_set <- "E_TLS"
geneset[geneset$gene%in%lccPFLTLSagainstLUNG$X,]$Gene_set <- "PFL_TLS"
geneset[geneset$gene%in%lccSFLTLSagainstLUNG$X,]$Gene_set <- "SFL_TLS"
geneset[geneset$gene%in%LSCC_allLUNG_DEGs,]$Gene_set <- "Lung"
geneset[geneset$gene%in%commonLSCC_allAGG_DEGs,]$Gene_set <- "all_TLS"
#geneset[geneset$gene=="SFTPB",]$Gene_set <- "Lung"

geneset <- geneset %>% mutate(Gene_set = factor(Gene_set, levels = c("Lung", "all_TLS", "Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS", "Other")))
table(geneset$Gene_set)

rows <- as.data.frame(geneset[,2])
rownames(rows) <- rownames(hm)
colnames(rows) <- "Gene_set"
rows2 <- rows %>% arrange(Gene_set)

# reorder the matrix based in the annotation
hm_ordered <- hm[rownames(rows2), ]
dim(hm_ordered)


#colon annotations
annotation <- data[,c("Tissue_Subtype3", "Patient_ID")]
#annotation <- data[,c("Tissue_Subtype3","Patient_Category", "Patient_ID")]


my_colour = list(Tissue_Subtype3 = c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2",
                              Alveoles = "#ff7f00",
                              LSCC = "#ffd700"
                              ),
                #Patient_Category = c(STS = "orchid1",
                 #     LTS = "steelblue4"),
                Patient_ID = c("32288" = "#CAB2D6",
                              "17776" = "#FDBF6F",
                              "24136" = "maroon",
                              "16690" = "skyblue2",
                              "38234" = "deeppink1",
                              "39160" = "palegreen2"
                              ),
                Gene_set = c(SFL_TLS = "#ff0000",
                              Tcell_TLS = "#6A33C2",
                              PFL_TLS = "#33a02c",
                             E_TLS = "#1F78C8",
                             Lung = "#ffd700",
                             all_TLS = "deeppink1",
                            Other = "grey"
                              )
    )


ph <- pheatmap::pheatmap(scale(hm), annotation_col = annotation, annotation_colors = my_colour)
range(scale(hm))

breaklist <- seq(-2, 4, by = 0.1) 
cols <- colorRampPalette(rev(brewer.pal(n=11, name = "RdYlBu")))(length(breaklist))

pdf("DESeq2_pheatmap_LSCC_totalTLSDEGs_LOQ3scaled_ward_0585lfcDEGs.pdf", width = 10, height = 10)
ph <- pheatmap::pheatmap(scale(hm), color = cols, breaks = breaklist, annotation_col = annotation, annotation_colors = my_colour,annotation_row = rows, cluster_cols = T, clustering_distance_cols = "euclidean", clustering_distance_rows = "euclidean", clustering_method = "ward.D", labels_col = "", cutree_cols = 4)
dev.off()

pdf("DESeq2_pheatmap_LSCC_totalTLSDEGs_LOQ3scaled_ward2.pdf", width = 10, height = 10)
ph <- pheatmap::pheatmap(scale(hm_ordered), color = cols, breaks = breaklist, annotation_col = annotation, annotation_row = rows, annotation_colors = my_colour, cluster_rows = F, clustering_distance_cols = "euclidean", clustering_method = "ward.D", labels_col = "", cutree_cols = 5)
dev.off()
#clustering_distance_cols = "euclidean", clustering_distance_rows = "euclidean", clustering_method = "ward.D",

```

# versus kidney
```{r load data}
cts <- as.matrix(m.rcc.end)
Meta_data_reorganised <- pD.rcc %>% mutate(Tissue_type = factor(Tissue_type))
unique(Meta_data_reorganised$Tissue_Subtype)

Meta_data_reorganised <- Meta_data_reorganised %>% mutate(Tissue_type2 = dplyr::recode(Tissue_type,
  "Tum" = "Kidney",
  "Kid" = "Kidney"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_type2)

dim(cts)

table(dds$Tissue_type2)
```

```{r DESeq2 normalisation TLS}
## set reference group for comparisons
dds$Tissue_type2 <- relevel(dds$Tissue_type2, ref = "TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results TLS}

resLFC <- lfcShrink(dds, coef="Tissue_type2_Kidney_vs_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_allTLSvsKidney_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

rccALLTLSvsKidneyDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.01]))

```

```{r colors}
c25 <- c(
  "dodgerblue2", "#E31A1C", # red
  "green4",
  "#6A3D9A", # purple
  "#FF7F00", # orange
  "black", "gold1",
  "skyblue2", "#FB9A99", # lt pink
  "palegreen2",
  "#CAB2D6", # lt purple
  "#FDBF6F", # lt orange
  "gray70", "khaki2",
  "maroon", "orchid1", "deeppink1", "blue1", "steelblue4",
  "darkturquoise", "green1", "yellow4", "yellow3",
  "darkorange4", "brown"
)
pie(rep(1, 25), col = c25)

 "#662965",
                  ifelse(rownames(data) %in% ID_W5, "#93CAEE",
                  ifelse(rownames(data) %in% ID_W6, "#1A58A8",
                  ifelse(rownames(data) %in% ID_W7, "#C19769",
                  ifelse(rownames(data) %in% ID_W8, "#D9E650",
                  ifelse(rownames(data) %in% ID_W9, "#417652",
                  ifelse(rownames(data) %in% ID_W10, "#00A94F",
                  ifelse(rownames(data) %in% ID_W11, "#A7D05C",
                  "#8F8F8F")))))))))))
                                                                                           
keyvals.colour[is.na(keyvals.colour)] <- 'black'

pal <- c("#1F78C8", "#ff0000", "#33a02c", "#6A33C2", "#ff7f00", 
        "#565656", "#FFD700", "#a6cee3", "#FB6496", "#b2df8a", 
        "#CAB2D6", "#FDBF6F", "#999999", "#EEE685", "#C8308C", 
        "#FF83FA", "#C814FA", "#0000FF", "#36648B", "#00E2E5", 
        "#00FF00", "#778B00", "#BEBE00", "#8B3B00", "#A52A3C")
```

```{r Volcano TLS vs kidney+RCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_allTLSvsKidney_LOQ3.csv")

Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -0.585 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 0.585 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Kidney'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
rccALLTLSagainstKID <- subset(Dat, expression =="Down")
rccKIDaganistALLTLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 100),
    xlim = c(-3,4),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_TLSvsKID_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 8, height = 8)

```

```{r extract transformed values}
vsd <- vst(dds, blind=FALSE)
#rld <- rlog(dds, blind=FALSE)

dim(assay(vsd))
DESeq2normexprALL <- t(assay(vsd))
#DESeq2normexpr <- as.data.frame(DESeq2normexprALL[,differentialrcc_allAGG_DEGs])

colnames(Meta_data_reorganised)
data <- Meta_data_reorganised[,c(1:4, 23:30, 35:38, 40, 44,47,48,50)]
data <- data %>%
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "RCC" = "Kidney",
  "Kidney" = "Kidney"
  )) %>%
  mutate(Tissue_Subtypefacet = factor(Tissue_Subtype2, levels = c("Kidney", "Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS"))) %>% 
  mutate(Tissue_Subtypecolor = factor(Tissue_Subtype2, levels = c("E_TLS", "SFL_TLS", "PFL_TLS", "Tcell_TLS", "Kidney")))
#data <- cbind(data, DESeq2normexpr)

colnames(data)
unique(data$Tissue_Subtype)
endcol <-ncol(data)



```

```{r, heatmap all TLS vs kidney, fig.width=6, fig.height=4}

#row_order <- data %>% mutate(rows = rownames(data)) %>% arrange(Tissue_Subtypefacet) %>% pull(rows)
#hm <- data[row_order,25:ncol(data)]
#annotation <- data[row_order,c("Tissue_Subtypefacet","Tumor_type")]
#annotation <- annotation %>% mutate(Tumor_type = factor(Tumor_type))
#rownames(annotation) <- row_order

selectedgenes <- sort(c(significant))
#selectedgenes <- sort(c(significant, "IGHG2", "IGHG3", "IGHG4"))
exclude <- c()
#exclude <- c("CIITA", "LSP1", "RASA3", "USP17L11", "TBC1D10C", "C7", "PTGDS", "SFRP2", "C3", "MGP", "AGPAT4", "SCN9A")
selectedgenes <- selectedgenes[!selectedgenes%in%exclude]


hm <- assay(vsd[selectedgenes,])
dim(hm)
#row annotation
geneset <- data.frame(genes = rownames(hm), Gene_set = "Other")
rownames(geneset) <- rownames(hm)

#recode gene set
geneset[geneset$gene%in%TcellDEGS,]$Gene_set <- "Tcell_TLS"
geneset[geneset$gene%in%EDEGS,]$Gene_set <- "E_TLS"
geneset[geneset$gene%in%PFLDEGS,]$Gene_set <- "PFL_TLS"
geneset[geneset$gene%in%SFLDEGS,]$Gene_set <- "SFL_TLS"
geneset[geneset$gene%in%RCC_allKidney_DEGs,]$Gene_set <- "Kidney"
geneset[geneset$gene%in%commonRCC_allAGG_DEGs,]$Gene_set <- "all_TLS"
#geneset[geneset$gene=="SFTPB",]$Gene_set <- "Kidney"

geneset <- geneset %>% mutate(Gene_set = factor(Gene_set, levels = c("Kidney", "all_TLS", "Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS", "Other")))
table(geneset$Gene_set)

rows <- as.data.frame(geneset[,2])
rownames(rows) <- rownames(hm)
colnames(rows) <- "Gene_set"
rows2 <- rows %>% arrange(Gene_set)

# reorder the matrix based in the annotation
hm_ordered <- hm[rownames(rows2), ]
dim(hm_ordered)


#colon annotations
annotation <- data[,c("Tissue_Subtype","Patient_ID")]
#annotation <- data[,c("Tissue_Subtype","Patient_Category", "Patient_ID")]


my_colour = list(Tissue_Subtype = c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2",
                              Kidney = "#ff7f00",
                              RCC = "#ffd700"
                              ),
               # Patient_Category = c(STS = "orchid1",
               #       LTS = "steelblue4"
               #         ),
                Patient_ID = c("r19101" = "#CAB2D6",
                              "r24784" = "#FDBF6F",
                              "r15660" = "maroon",
                              "r8527" = "skyblue2",
                              "r30616" = "deeppink1",
                              "r18773" = "palegreen2"
                                ),
                      Gene_set = c(SFL_TLS = "#ff0000",
                              Tcell_TLS = "#6A33C2",
                              PFL_TLS = "#33a02c",
                             E_TLS = "#1F78C8",
                             Kidney = "#ffd700",
                             all_TLS = "deeppink1",
                            Other = "grey"
                              )
    )


ph <- pheatmap::pheatmap(scale(hm), annotation_col = annotation, annotation_colors = my_colour)
range(scale(hm))

breaklist <- seq(-2, 4, by = 0.1) 
cols <- colorRampPalette(rev(brewer.pal(n=11, name = "RdYlBu")))(length(breaklist))

pdf("DESeq2_pheatmap_RCC_totalTLSDEGs_LOQ3scaled_ward_lfc0585DEGs.pdf", width = 10, height = 10)
ph <- pheatmap::pheatmap(scale(hm), color = cols, breaks = breaklist, annotation_col = annotation, annotation_row = rows, annotation_colors = my_colour, cluster_cols = T, labels_col = "", cutree_cols = 3)
dev.off()

pdf("DESeq2_pheatmap_RCC_totalTLSDEGs_LOQ3scaled_ward3.pdf", width = 18, height = 10)
ph <- pheatmap::pheatmap(scale(hm_ordered), color = cols, breaks = breaklist, annotation_col = annotation, annotation_row = rows, annotation_colors = my_colour, cluster_cols = T,cluster_rows = F, clustering_distance_cols = "euclidean", clustering_distance_rows = "euclidean", clustering_method = "ward.D",  cutree_cols = 6)
dev.off()
#clustering_distance_cols = "euclidean", clustering_distance_rows = "euclidean", clustering_method = "ward.D",

```


## TLS DEGs shared in LSCC and RCC
```{r}
lccALLTLSagainstLUNG
rccALLTLSagainstKID

sharedDEGs <- sort(Reduce(intersect,list(lccALLTLSagainstLUNG$X, rccALLTLSagainstKID$X)))

PARENCHYMEDEGs <- sort(unique(c(lsccLUNGaganistALLTLS$X, rccKIDaganistALLTLS$X)))


```

```{r load data}
cts <- as.matrix(m.tlstum.end)
Meta_data_reorganised <- pD.TLStum %>% mutate(Tissue_type = factor(Tissue_type))
unique(Meta_data_reorganised$Tissue_type)

Meta_data_reorganised <- Meta_data_reorganised %>% mutate(Tissue_type2 = dplyr::recode(Tissue_type,
  "Tum" = "Parenchyma",
  "Alv" = "Parenchyma",
  "Kid" = "Parenchyma"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_type2)

dim(cts)

table(dds$Tissue_type2)
```

```{r DESeq2 normalisation total TLS}
## set reference group for comparisons
dds$Tissue_type2 <- relevel(dds$Tissue_type2, ref = "TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r extract transformed values}
vsd <- vst(dds, blind=FALSE)
dim(assay(vsd))
DESeq2normexprALL <- t(assay(vsd))
DESeq2normexpr <- as.data.frame(DESeq2normexprALL[,sharedDEGs])

unique(Meta_data_reorganised$Tissue_type)

colnames(Meta_data_reorganised)
data <- Meta_data_reorganised[,c(1:4, 23:30, 35:38, 40, 44,47,48,50)]
data <- data %>%
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "LSCC" = "Lung",
  "Alveoles" = "Lung",
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS",
  "RCC" = "Kidney"
  )) %>%
  mutate(Tissue_Subtypecolor = factor(Tissue_Subtype2, levels = c("E_TLS", "SFL_TLS", "PFL_TLS", "Tcell_TLS", "Lung", "Kidney", "LN_T", "LN_B", "LN_GC"))) %>% 
  mutate(Tissue_type2 = dplyr::recode(Tissue_type,
  "Alv" = "N",
  "Kid" = "N",
  "Tum" = "T"
  )) %>% 
  mutate(Tissue_type2 = factor(Tissue_type2, levels = c("N", "T", "TLS", "LN")))

data1 <- cbind(data, DESeq2normexpr)

colnames(data1)
unique(data1$Tissue_Subtype2)
endcol <-ncol(data1)



```

#Expression dot plots of total TLS DEGs in TLS subtypes
```{r colors}
pal <- c("#1F78C8", "#ff0000", "#33a02c", "#6A33C2", "#ff7f00", 
        "#565656", "#FFD700", "#a6cee3", "#FB6496", "#b2df8a", 
        "#CAB2D6", "#FDBF6F", "#999999", "#EEE685", "#C8308C", 
        "#FF83FA", "#C814FA", "#0000FF", "#36648B", "#00E2E5", 
        "#00FF00", "#778B00", "#BEBE00", "#8B3B00", "#A52A3C")

dotplot_colour =  c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2",
                              Kidney = "#ff7f00",
                              Lung = "#FFD700",
                              LN_B = "#a6cee3",
                              LN_GC = "#FB6496",
                              LN_T = "#FF83FA"
                              )
    

```

```{r DEGs in AOI classes by tissue type}

i=29
for (i in 24:endcol) {
 pp_prefix <- "Dotplot_DESeq2"
var <- "TLSTUMdata"
 plot = colnames(data1[i])
ggplot(data = data1, aes(x = Tissue_type2 , y = data1[,i])) +
    geom_boxplot(alpha = 1, width = 0.4, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, width = 0.05, size=3, pch=21, aes(fill=Tissue_Subtypecolor), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
  #scale_color_aaas() +
    #scale_y_continuous(trans = "log10") +
  scale_fill_manual(values=dotplot_colour) +
    facet_wrap(~Tumor_type, nrow = 2,) +
    stat_compare_means(comparisons=list(c("N", "T"), c("T", "TLS"), c("N", "TLS")), size = 5) +
    stat_compare_means(comparisons=list(c("TLS", "LN")), size = 5) + 
  labs(x = "Tissue type",
       y= "Normalised expression",
       fill = "AOI subtype",
      title = colnames(data1[i])
      ) +
  #theme_pubr(border = TRUE) +
        theme_light() + 
    theme(text=element_text(size=20),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          strip.text = element_text(colour='black'),
        #axis.text.x = element_text(angle = 55, hjust = 1, size=14),
          legend.title = element_text(size=14),
        legend.text = element_text(size=14), 
        legend.position = "none",
          plot.title = element_text(size=16, hjust = 0.5))

fn <- sprintf("%s_%s_%s_2.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 3, height = 6)



 }
```

#Pathway analysis
```{r}
Dat  <- read.csv("DESeq2_allTLSTUM_allTLSvsParenchyma_LOQ3.csv")
df <- subset(Dat, X%in%c(sharedDEGs))

# we want the log2 fold change 
original_gene_list <- df$log2FoldChange
original_gene_list <- original_gene_list*-1

# name the vector
names(original_gene_list) <- df$X

# omit any NA values 
gene_list<-na.omit(original_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)
gene_list <- 2^gene_list
```

```{r clusterprofiler}
gse <- gseGO(geneList=gene_list, 
             ont ="BP", 
             keyType = "SYMBOL", 
             #nPerm = 10000, 
             minGSSize = 10, 
             maxGSSize = 100, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             #scoreType = "pos",
             OrgDb = org.Hs.eg.db, 
             pAdjustMethod = "BH")


dotplot(gse, showCategory=10, split=".sign") + facet_grid(.~.sign)
ggsave(filename = "pathwayGOdotplot_sharedallTLSgenes_lFC0585_fdr005.pdf", width = 9, height = 6)

gseaplot(gse, by = "all", title = gse$Description[7], geneSetID = 1)
ggsave(filename = "pathwayGOcation homeostasis_TLSmed_woCryoTLS_lFC0485_fdr008.pdf", width = 6, height = 6)

ridgeplot(gse) + labs(x = "enrichment distribution")
ggsave(filename = "pathwayGOridgeplot_sharedallTLS_lFC0585_fdr005.pdf", width = 12, height = 12)

```

```{r GSEA}

pw1 <- gmtPathways("c2.all.v7.3.symbols.gmt")
pw2 <- gmtPathways("c5.all.v7.3.symbols.gmt")
pw3 <- gmtPathways("c6.all.v7.3.symbols.gmt")
pw.all <- c(pw1, pw2, pw3)

fg1 <- fgsea(pw.all, sort(gene_list), minSize = 5, maxSize = 2000)

data.table(fg1)
setkey(fg1, padj)
View(fg1[fg1$pval < 0.05, ])

plotEnrichment(pw.all[["GOBP_HUMORAL_IMMUNE_RESPONSE"]], sort(gene_list))
gseatab1 <- plotGseaTable(pw.all[fg1[fg1$padj < 0.05, pathway]], gene_list, fg1, gseaParam = 0.5, render = FALSE)
pdf("GSEA_sharedallTLS.pdf", width = 12, height = 12)
plot(gseatab1)
dev.off()
```

```{r KEGG}
#for KEGG

Dat  <- read.csv("DESeq2_allTLSTUM_allTLSvsParenchyma_LOQ3.csv")
df <- subset(Dat, X%in%c(sharedDEGs)) #PARENCHYMEDEGs

# we want the log2 fold change 
original_gene_list <- df$log2FoldChange
original_gene_list <- original_gene_list*-1

# name the vector
names(original_gene_list) <- df$X

# omit any NA values 
gene_list<-na.omit(original_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)

# Convert gene IDs for gseKEGG function
# We will lose some genes here because not all IDs will be converted
ids<-bitr(names(gene_list), fromType = "SYMBOL", toType = "ENTREZID", OrgDb=org.Hs.eg.db)
 # remove duplicate IDS (here I use "ENTREZID", but it should be whatever was selected as keyType)
dedup_ids = ids[!duplicated(ids[c("SYMBOL")]),]

# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
df2 = df[df$X %in% dedup_ids$SYMBOL,]

# Create a new column in df2 with the corresponding ENTREZ IDs
df2$Y = dedup_ids$ENTREZID

# Create a vector of the gene unuiverse
kegg_gene_list <- df2$log2FoldChange


# Name vector with ENTREZ ids
names(kegg_gene_list) <- df2$Y

# omit any NA values 
kegg_gene_list<-na.omit(kegg_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

kegg_gene_list <- kegg_gene_list *-1
kegg_gene_listlinear <- 2^kegg_gene_list



```

```{r KEGG plots}
###check https://bioc.ism.ac.jp/packages/3.7/bioc/vignettes/enrichplot/inst/doc/enrichplot.html

#If you use DOSE in published research, please cite:
#Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609

#library(DOSE)
#install.packages("ggnewscale")

de <- names(kegg_gene_listlinear)[abs(kegg_gene_listlinear) > 1.5]
ego <- enrichGO(de, OrgDb = "org.Hs.eg.db", ont="BP", readable=TRUE) #pool = TRUE -> when useing ont="ALL"
##ont	: One of "BP", "MF", and "CC" subontologies, or "ALL" for all three.
dotplot(ego)
#ggsave(filename = "pathwayMFdotplotFDRnone_TLSmed_wCryoTLS_unique_lFC0585_fdr005.pdf", width = 7, height = 3)

go <- enrichGO(de, OrgDb = "org.Hs.eg.db", ont="ALL", readable=TRUE, pvalueCutoff = 0.05,
  pAdjustMethod = "BH")
dotplot(go, split="ONTOLOGY") + facet_grid(ONTOLOGY~., scale="free")
dotplot(go, split="ONTOLOGY") + facet_grid(.~ONTOLOGY)

ggsave(filename = "pathwayALLdotplotBH005_sharedallTLS_lFC0585_fdr005.pdf", width = 10, height = 7)

barplot(ego, showCategory=10)
ggsave(filename = "pathwayBPbarplotBH005_sharedallTLS_lFC0585_fdr005.pdf", width = 10, height = 7)


##Interaction maps
#https://rdrr.io/bioc/enrichplot/man/cnetplot.html
##\item{layout}{Layout of the map, e.g. 'star', 'circle', 'gem', 'dh', 'graphopt', 'grid', 'mds', 'randomly', 'fr', 'kk', 'drl' or 'lgl'.}
# remove redundent GO terms
#ego2 <- simplify(ego)
cnetplot(ego, foldChange=kegg_gene_listlinear, showCategory=10, layout="dh")
ggsave(filename = "pathwayBPinteractionBH005_TLSmed_woCryo_lFC0585_fdr005.pdf", width = 5, height = 8)

cnetplot(ego, foldChange=kegg_gene_listlinear, circular = TRUE, colorEdge = TRUE, showCategory=10, cex_label_category = 0.7)
ggsave(filename = "pathwayBPinteractionCircBH005_TLSmed_woCryo_lFC0585_fdr005.pdf", width = 7, height = 4)



#Disease associated pathways:
edo <- enrichDGN(names(kegg_gene_list), 
                 pvalueCutoff = 0.05,
  pAdjustMethod = "fdr",
  minGSSize = 10,
  maxGSSize = 500,
  qvalueCutoff = 0.05,
  readable = FALSE)

barplot(edo, showCategory=20)
ggsave(filename = "pathwayDGNbarplotFDR005_TLSmed_wCryo_unique_lFC0585_fdr005.pdf", width = 10, height = 6)

edo2 <- gseNCG(kegg_gene_list, #nPerm=10000
               minGSSize = 3,
  maxGSSize = 500,
  pvalueCutoff = 0.05,
  pAdjustMethod = "none"
               )
p1 <- dotplot(edo, showCategory=10) + ggtitle("dotplot for ORA")
ggsave(filename = "pathwayORAdotplotFDR005_CryoTLS_lFC0485_fdr005.pdf", width = 10, height = 6)

p2 <- dotplot(edo2, showCategory=10) + ggtitle("dotplot for GSEA")
plot_grid(p1, p2, ncol=2)
```



## TLS maturation sequence in LSCC
```{r load data}
cts <- as.matrix(m.lscc.end)
Meta_data_reorganised <- pD.lscc %>% mutate(Tissue_Subtype = factor(Tissue_Subtype))
unique(Meta_data_reorganised$Tissue_Subtype)
```

# Tcell TLS vs Other TLS
```{r data for T cell-TLS agains other TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% 
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "LSCC" = "Lung",
  "Alveoles" = "Lung",
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "SFL_TLS" = "TLS",
  "PFL_TLS" = "TLS",
  "E_TLS" = "TLS"
  ))%>% 
  mutate(Tissue_Subtype4 = dplyr::recode(Tissue_Subtype,
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype4)

dim(cts)

table(dds$Tissue_Subtype4)
```

```{r DESeq2 normalisation Tcell_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "Tcell_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results Tcell vs Other TLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_TLS_vs_Tcell_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_TcellvsOtherTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

```

```{r Volcano Tcell vs Other TLS in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_TcellvsOtherTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in Tcell_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Other_TLS'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
lccTCELLSagainstOtherTLS <- subset(Dat, expression =="Down")
lccOtherTLSaganistTCELL <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
   ylim = c(0, 3),
   xlim = c(-2,2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_TcellTLSvsOtherTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 results Tcell vs Lung}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_Lung_vs_Tcell_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_TcellvsLung_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

```

```{r Volcano Tcell vs Lung in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_TcellvsLung_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -0.585 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 0.585 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in Tcell_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Lung'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
#lccTCELLSagainstLung <- subset(Dat, expression =="Down")
#lccLungaganistTCELL <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant, "LIMD2"),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 20),
    xlim = c(-4,5),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_TcellTLSvsLung_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```


# Tcell TLS vs mTLS
```{r data for T cell-TLS agains other TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% 
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "LSCC" = "Lung",
  "Alveoles" = "Lung",
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "SFL_TLS" = "TLS",
  "PFL_TLS" = "TLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype3)

dim(cts)

table(dds$Tissue_Subtype3)
```

```{r DESeq2 normalisation Tcell_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "Tcell_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results Tcell vs PSFL TLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_TLS_vs_Tcell_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_TcellvsPSFLTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccTcellvsPSFLTLSDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.05]))

```

```{r Volcano Tcell vs PSFL TLS in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_TcellvsPSFLTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in Tcell_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in PFL+SFL'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
lccTCELLSagainstPSFLTLS <- subset(Dat, expression =="Down")
lccPSFLTLSaganistTCELL <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
   #ylim = c(0, 3),
   #xlim = c(-2,2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_TcellTLSvsPSFLTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

#E_TLS
```{r data for E-TLS agains other TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% 
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "LSCC" = "Lung",
  "Alveoles" = "Lung",
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "SFL_TLS" = "TLS",
  "PFL_TLS" = "TLS"
  )) %>% 
  mutate(Tissue_Subtype4 = dplyr::recode(Tissue_Subtype,
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype4)

dim(cts)

table(dds$Tissue_Subtype4)
```

```{r DESeq2 normalisation Tcell_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "E_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results E vs Tcell}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_Tcell_TLS_vs_E_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_ETLSvsTcellTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccTcellvsOtherTLSDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.05]))

```

```{r Volcano E vs Tcell in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_ETLSvsTcellTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in E_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Tcell_TLS'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
lccETLSagainstTCELL <- subset(Dat, expression =="Down")
lccTCELLaganistETLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 3),
    xlim = c(-2,2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_ETLSvsTcellTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 results E vs PSFL}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_TLS_vs_E_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_ETLSvsotherTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccTcellvsOtherTLSDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.05]))

```

```{r Volcano E vs PSFL in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_ETLSvsotherTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in E_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in PFL+SFL'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
lccETLSagainstPSFLTLS <- subset(Dat, expression =="Down")
lccPSFTLSaganistETLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 20),
    xlim = c(-1,2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_ETLSvsPSFLTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 results E vs Lung}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_Lung_vs_E_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_ETLSvsLung_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccEvsLungDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.05]))

```

```{r Volcano E vs Lung in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_ETLSvsLung_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -0.585 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 0.585 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in E_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Lung'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
#lccETLSagainstLung <- subset(Dat, expression =="Down")
#lccLungaganistETLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 80),
    xlim = c(-4,5),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_ETLSvsLUNG_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 normalisation Tcell_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype4 <- relevel(dds$Tissue_Subtype4, ref = "E_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results E vs Alveoles}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype4_Alveoles_vs_E_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_ETLSvsAlveoles_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

# Tcell+E_TLS
```{r data for E-TLS agains other TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% 
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "LSCC" = "Lung",
  "Alveoles" = "Lung",
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "SFL_TLS" = "mTLS",
  "PFL_TLS" = "mTLS",
  "E_TLS" = "eTLS",
  "Tcell_TLS" = "eTLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype3)

dim(cts)

table(dds$Tissue_Subtype3)
```

```{r DESeq2 normalisation Tcell_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "eTLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results ETCELL vs PSFL}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_mTLS_vs_eTLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_ETcellvsPSFL_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccETcellvsPSFLTLSDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.05]))

```

```{r Volcano ETCELL vs PSFL in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_ETcellvsPSFL_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in Tcell+E'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in PFL+SFL'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
lccETLSagainstTCELL <- subset(Dat, expression =="Down")
lccTCELLaganistETLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- c(as.vector(Df$X), "CCL21")

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 20),
    xlim = c(-1,2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_ETcellvsPSFLTLS_volcano_lfc0265_fdr005_CCL21.pdf")
ggsave(file_name, width = 6, height = 6)

```

#PFL_TLS
```{r data for PFL-TLS agains other TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% 
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "LSCC" = "Lung",
  "Alveoles" = "Lung",
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "E_TLS" = "eTLS",
  "Tcell_TLS" = "eTLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype3)

dim(cts)

table(dds$Tissue_Subtype3)
```

```{r DESeq2 normalisation PFL_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "PFL_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results PFL vs eTLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_eTLS_vs_PFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_PFLvsETCELL_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccPFLvsETcellDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.05]))

```

```{r Volcano PFL vs eTLS in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_PFLvsETCELL_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in PFL_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Tcell+E'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
lccPFLagainstETCELL <- subset(Dat, expression =="Down")
lccETCELLaganistPFL <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 3),
    xlim = c(-2,2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_PFLvsETcell_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 results PFL vs SFL TLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_SFL_TLS_vs_PFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_PFLTLSvsSFLTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccPFLvsSFLTLSDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.05]))

```

```{r Volcano PFL vs SFL in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_PFLTLSvsSFLTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in PFL_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in SFL_TLS'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
lccPFLTLSagainstSFLTLS <- subset(Dat, expression =="Down")
lccSFTLSaganistPFLTLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 8),
    xlim = c(-1,1),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_PFLTLSvsSFLTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 results PFL vs Lung}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_Lung_vs_PFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_PFLTLSvsLung_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccPFLvsLungDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.05]))

```

```{r Volcano PFL vs Lung in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_PFLTLSvsLung_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -0.585 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 0.585 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in PFL_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Lung'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
#lccPFLTLSagainstLung <- subset(Dat, expression =="Down")
#lccLungaganistPFLTLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 20),
    xlim = c(-4,5),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_PFLTLSvsLUNG_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

#SFL_TLS
```{r data for SFL-TLS agains other TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% 
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "LSCC" = "Lung",
  "Alveoles" = "Lung",
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "E_TLS" = "TLS",
  "Tcell_TLS" = "TLS",
  "PFL_TLS" = "TLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype3)

dim(cts)

table(dds$Tissue_Subtype3)
```

```{r DESeq2 normalisation Tcell_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "SFL_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results SFL vs eTLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_TLS_vs_SFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_SFLvsOtherTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccSFLvsOtherTLSDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.05]))

```

```{r Volcano SFL vs eTLS in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_SFLvsOtherTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in SFL_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Other_TLS'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
lccSFLagainstOtherTLS <- subset(Dat, expression =="Down")
lccOtherTLSaganistSFL <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 20),
    xlim = c(-2,2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 5.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 16,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_SFLvsOtherTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 7, height = 10)







```

```{r DESeq2 results SFL vs Lung}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_Lung_vs_SFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_SFLTLSvsLung_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

lsccSFLvsLungDEGs <- sort(as.character(resOrdered$Gene[resOrdered$padj < 0.05]))

```

```{r Volcano SFL vs Lung in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_lscc_SFLTLSvsLung_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -0.585 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 0.585 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in SFL_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Lung'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
#lccSFLTLSagainstLung <- subset(Dat, expression =="Down")
#lccLungaganistSFLTLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 80),
    xlim = c(-4,5),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_SFLTLSvsLUNG_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r data for SFL-TLS agains E-TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% 
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "LSCC" = "Lung",
  "Alveoles" = "Lung",
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "E_TLS" = "TLS",
  "Tcell_TLS" = "TLS",
  "PFL_TLS" = "TLS"
  ))%>% 
  mutate(Tissue_Subtype4 = dplyr::recode(Tissue_Subtype,
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype4)

dim(cts)

table(dds$Tissue_Subtype4)
```

```{r DESeq2 normalisation Tcell_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype4 <- relevel(dds$Tissue_Subtype4, ref = "SFL_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results SFL vs eTLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype4_E_TLS_vs_SFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_lscc_SFLvsETLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

```

#List DEGs of maturation stages in LSCC
```{r}
lccTCELLSagainstOtherTLS
lccTCELLaganistETLS

lccOtherTLSaganistTCELL
lccETLSagainstTCELL 


lccETLSagainstPSFLTLS 
lccPSFTLSaganistETLS 

lccSFLagainstOtherTLS 
lccOtherTLSaganistSFL 

Reduce(intersect,list(lccPSFTLSaganistETLS$X, lccSFLagainstOtherTLS$X))
setdiff(lccSFLagainstOtherTLS$X, lccPSFTLSaganistETLS$X)

TLSmaturationDEGs <- sort(unique(c(lccTCELLSagainstOtherTLS$X, lccTCELLaganistETLS$X,lccOtherTLSaganistTCELL$X, lccETLSagainstTCELL$X, lccETLSagainstPSFLTLS$X, lccPSFTLSaganistETLS$X, lccSFLagainstOtherTLS$X, lccOtherTLSaganistSFL$X)))

#AGAINST LUNG
lccSFLTLSagainstLung 
#lccLungaganistSFLTLS 
lccTCELLSagainstLung 
#lccLungaganistTCELL 
lccETLSagainstLung 
#lccLungaganistETLS 
lccPFLTLSagainstLung 
#lccLungaganistPFLTLS 

```

```{r total TLS DEGs against lung}
TLSDEG_foldchange_Lung <- merge(lccTCELLSagainstLung[,c(1,3)],lccETLSagainstLung[,c(1,3)], by = "X", all = TRUE)
TLSDEG_foldchange_Lung <- merge(TLSDEG_foldchange_Lung, lccPFLTLSagainstLung[,c(1,3)], by = "X", all = TRUE)
TLSDEG_foldchange_Lung <- merge(TLSDEG_foldchange_Lung, lccSFLTLSagainstLung[,c(1,3)], by = "X", all = TRUE)

colnames(TLSDEG_foldchange_Lung) <- c("Gene", "Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS")
TLSDEG_foldchange_LungT <- as.data.frame(t(TLSDEG_foldchange_Lung[,c(2:5)]))
colnames(TLSDEG_foldchange_LungT) <- TLSDEG_foldchange_Lung$Gene
TLSDEG_foldchange_LungT <- (TLSDEG_foldchange_LungT *-1) 
TLSDEG_foldchange_LungT$TLStype <- as.factor(colnames(TLSDEG_foldchange_Lung[,2:5]))

endcol <- ncol(TLSDEG_foldchange_LungT) -1

#setdiff(lccALLTLSagainstLUNG$X, TLSDEG_foldchange_Lung$Gene)
```

```{r total TLS DEG fold change barplots}
dotplot_colour <- c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2"
                              )
    

i=2
for (i in 2:endcol) { 
pp_prefix <- "Barplot_DESeq2"
var <- "DEGfoldchange_inTLStypes"
 plot = colnames(TLSDEG_foldchange_LungT[i])
 
ggplot(data = TLSDEG_foldchange_LungT, aes(x = factor(TLStype, levels = c("Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS")), y = TLSDEG_foldchange_LungT[,i], fill = TLStype)) +
    geom_bar(stat="identity", position = 'identity') +
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5), limits=c(0.9, NA), oob=rescale_none, expand = expansion(mult = c(0.1, 0.1))) +
  #scale_color_aaas(values=dotplot_colour) +
   scale_fill_manual(values=dotplot_colour) +
    labs(x = "TLS type",
       y= bquote(~Log[2]~ 'FC versus lung'),
      title = colnames(TLSDEG_foldchange_LungT[i])
      ) +
  #theme_pubr(border = TRUE) +
    theme_light() + 
    theme(text=element_text(size=22),
          plot.title = element_text(hjust = 0.5),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          axis.text.x = element_text(angle = 45, hjust = 1),
          legend.position = "none",
          strip.text = element_text(colour='black'))

fn <- sprintf("%s_%s_%s.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 4, height = 5)
}
```

```{r maturation TLS DEGs against lung}
SFL  <- read.csv("DESeq2_lscc_SFLTLSvsLung_LOQ3.csv")
PFL  <- read.csv("DESeq2_lscc_PFLTLSvsLung_LOQ3.csv")
E  <- read.csv("DESeq2_lscc_ETLSvsLung_LOQ3.csv")
TCELL  <- read.csv("DESeq2_lscc_TcellvsLung_LOQ3.csv")

TLSmaturationDEGs

TLSmaturDEGs_foldchnage_Lung <- merge(TCELL[,c(1,3)],E[,c(1,3)], by = "X", all = TRUE)
TLSmaturDEGs_foldchnage_Lung <- merge(TLSmaturDEGs_foldchnage_Lung, PFL[,c(1,3)], by = "X", all = TRUE)
TLSmaturDEGs_foldchnage_Lung <- merge(TLSmaturDEGs_foldchnage_Lung, SFL[,c(1,3)], by = "X", all = TRUE)
colnames(TLSmaturDEGs_foldchnage_Lung) <- c("Gene", "Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS")
TLSmaturDEGs_foldchnage_Lung_sel <- TLSmaturDEGs_foldchnage_Lung[TLSmaturDEGs_foldchnage_Lung$Gene%in%TLSmaturationDEGs,]

uniquelyTLSmaturationDEGs <- setdiff(TLSmaturDEGs_foldchnage_Lung_sel$Gene, TLSDEG_foldchange_Lung$Gene)
TLSmaturDEGs_foldchnage_Lung_sel2 <- TLSmaturDEGs_foldchnage_Lung[TLSmaturDEGs_foldchnage_Lung$Gene%in%uniquelyTLSmaturationDEGs,]


TLSmaturDEGs_foldchnage_Lung_sel2T <- as.data.frame(t(TLSmaturDEGs_foldchnage_Lung_sel2[,c(2:5)]))
colnames(TLSmaturDEGs_foldchnage_Lung_sel2T) <- TLSmaturDEGs_foldchnage_Lung_sel2$Gene
TLSmaturDEGs_foldchnage_Lung_sel2T <- (TLSmaturDEGs_foldchnage_Lung_sel2T *-1) 
TLSmaturDEGs_foldchnage_Lung_sel2T$TLStype <- as.factor(colnames(TLSmaturDEGs_foldchnage_Lung_sel2[,2:5]))

endcol <- ncol(TLSmaturDEGs_foldchnage_Lung_sel2T) -1

```

```{r total TLS DEG fold change barplots}
dotplot_colour <- c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2"
                              )
    

i=2
for (i in 2:endcol) { 
pp_prefix <- "Barplot_DESeq2"
var <- "DEGfoldchange_inTLStypes"
 plot = colnames(TLSmaturDEGs_foldchnage_Lung_sel2T[i])
 
ggplot(data = TLSmaturDEGs_foldchnage_Lung_sel2T, aes(x = factor(TLStype, levels = c("Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS")), y = TLSmaturDEGs_foldchnage_Lung_sel2T[,i], fill = TLStype)) +
    geom_bar(stat="identity", position = 'identity') +
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5), limits=c(NA, NA), expand = expansion(mult = c(0.1, 0.1))) +
  #scale_color_aaas(values=dotplot_colour) +
   scale_fill_manual(values=dotplot_colour) +
    labs(x = "TLS type",
       y= bquote(~Log[2]~ 'FC versus lung'),
      title = colnames(TLSmaturDEGs_foldchnage_Lung_sel2T[i])
      ) +
  #theme_pubr(border = TRUE) +
    theme_light() + 
    theme(text=element_text(size=22),
          plot.title = element_text(hjust = 0.5),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          axis.text.x = element_text(angle = 45, hjust = 1),
          legend.position = "none",
          strip.text = element_text(colour='black'))

fn <- sprintf("%s_%s_%s.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 4, height = 5)
}
```


##TLS maturation sequence in RCC
# versus kidney
```{r load data}
cts <- as.matrix(m.rcc.end)
Meta_data_reorganised <- pD.rcc %>% mutate(Tissue_type = factor(Tissue_type))
unique(Meta_data_reorganised$Tissue_Subtype)
```

# Tcell TLS vs Other TLS
```{r T cell vs Other TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "RCC" = "Kidney",
  "Kidney" = "Kidney"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "SFL_TLS" = "TLS",
  "PFL_TLS" = "TLS",
  "E_TLS" = "TLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype3)

dim(cts)

table(dds$Tissue_Subtype3)
```

```{r DESeq2 normalisation Tcell TLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "Tcell_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results T cell TLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_TLS_vs_Tcell_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_TCELLTLSvsOtherTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

```{r Volcano Tcell vs Other TLS in RCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_TCELLTLSvsOtherTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in Tcell_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Other_TLS'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
rccTCELLSagainstOtherTLS <- subset(Dat, expression =="Down")
rccOtherTLSaganistTCELL <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
   ylim = c(0, 15),
   xlim = c(-2,2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_TcellTLSvsOtherTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 results Tcell vs Kidney}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_Kidney_vs_Tcell_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_TcellvsKidney_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

```

```{r Volcano Tcell vs Kidney}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_TcellvsKidney_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -0.585 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 0.585 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in Tcell_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Kidney'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
#rccTCELLSagainstKidney <- subset(Dat, expression =="Down")
#rccKidneyaganistTCELL <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 80),
    xlim = c(-4,5),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_TcellTLSvsKidney_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

#E_TLS vs other TLS
```{r data for E-TLS agains other TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "RCC" = "Kidney",
  "Kidney" = "Kidney"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "SFL_TLS" = "TLS",
  "PFL_TLS" = "TLS"
  )) 

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype)

dim(cts)

table(dds$Tissue_Subtype)
```

```{r DESeq2 normalisation Tcell_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "E_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results E vs Tcell}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_Tcell_TLS_vs_E_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_ETLSvsTcellTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

```

```{r Volcano E vs Tcell in RCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_ETLSvsTcellTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in E_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Tcell_TLS'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
rccETLSagainstTCELL <- subset(Dat, expression =="Down")
rccTCELLaganistETLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 7),
    xlim = c(-2,2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_ETLSvsTcellTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 results E vs PSFL}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_TLS_vs_E_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_ETLSvsotherTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

```{r Volcano E vs PSFL in RCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_ETLSvsotherTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in E_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in PFL+SFL'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
rccETLSagainstPSFLTLS <- subset(Dat, expression =="Down")
rccPSFTLSaganistETLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 20),
    xlim = c(-1,1.2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_ETLSvsPSFLTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 results E vs Kidney}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_Kidney_vs_E_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_ETLSvsKidney_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

```{r Volcano E vs Kidney in RCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_ETLSvsKidney_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -0.585 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 0.585 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in E_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Kidney'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
#rccETLSagainstKidney <- subset(Dat, expression =="Down")
#rccKidneyaganistETLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 80),
    xlim = c(-4,5),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_ETLSvsKIDNEY_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 normalisation E_TLS agains normal kidney}
## set reference group for comparisons
dds$Tissue_Subtype <- relevel(dds$Tissue_Subtype, ref = "E_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results E vs Tcell}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype_Kidney_vs_E_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_ETLSvsnormalKidney_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

```

# Tcell+E_TLS
```{r data for E-TLS+Tcell agains PSFL TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "RCC" = "Kidney",
  "Kidney" = "Kidney"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "SFL_TLS" = "mTLS",
  "PFL_TLS" = "mTLS",
  "E_TLS" = "eTLS",
  "Tcell_TLS" = "eTLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype3)

dim(cts)

table(dds$Tissue_Subtype3)
```

```{r DESeq2 normalisation eTLS agains mTLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "eTLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results ETCELL vs PSFL}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_mTLS_vs_eTLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_ETcellvsPSFL_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

```{r Volcano ETCELL vs PSFL in lSCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_ETcellvsPSFL_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in Tcell+E'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in PFL+SFL'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
rccETCELLTLSagainstPSFL <- subset(Dat, expression =="Down")
rccPSFLaganistETCELLTLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- c(as.vector(Df$X))

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 26),
    xlim = c(-2,1.5),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_ETcellvsPSFLTLS_volcano_lfc0265_fdr005_2.pdf")
ggsave(file_name, width = 6, height =6)

```

#PFL_TLS
```{r data for PFL-TLS agains other TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "RCC" = "Kidney",
  "Kidney" = "Kidney"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "E_TLS" = "eTLS",
  "Tcell_TLS" = "eTLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype3)

dim(cts)

table(dds$Tissue_Subtype3)
```

```{r DESeq2 normalisation PFL_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "PFL_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results PFL vs eTLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_eTLS_vs_PFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_PFLvsETCELL_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

```{r Volcano PFL vs eTLS in RCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_PFLvsETCELL_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in PFL_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Tcell+E'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
rccPFLagainstETCELL <- subset(Dat, expression =="Down")
rccETCELLaganistPFL <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 5),
    xlim = c(-2,1),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_PFLvsETcell_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 results PFL vs SFL TLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_SFL_TLS_vs_PFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_PFLTLSvsSFLTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

```{r Volcano PFL vs SFL in RCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_PFLTLSvsSFLTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in PFL_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in SFL_TLS'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
rccPFLTLSagainstSFLTLS <- subset(Dat, expression =="Down")
rccSFTLSaganistPFLTLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 5),
    xlim = c(-2,1),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_PFLTLSvsSFLTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 results PFL vs Kidney}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_Kidney_vs_PFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_PFLTLSvsKidney_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

```{r Volcano E vs Kidney in RCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_PFLTLSvsKidney_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -0.585 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 0.585 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in PFL_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Kidney'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
#rccPFLTLSagainstKidney <- subset(Dat, expression =="Down")
#rccKidneyaganistPFLTLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 25),
    xlim = c(-4,5),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_PFLTLSvsKID_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

#SFL_TLS
```{r data for SFL-TLS agains other TLS}
Meta_data_reorganised <- Meta_data_reorganised %>% mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "RCC" = "Kidney",
  "Kidney" = "Kidney"
  )) %>% 
  mutate(Tissue_Subtype3 = dplyr::recode(Tissue_Subtype2,
  "E_TLS" = "TLS",
  "Tcell_TLS" = "TLS",
  "PFL_TLS" = "TLS"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype)

dim(cts)

table(dds$Tissue_Subtype)
```

```{r DESeq2 normalisation SFL_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype3 <- relevel(dds$Tissue_Subtype3, ref = "SFL_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results SFL vs eTLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_TLS_vs_SFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_SFLvsOtherTLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

```{r Volcano SFL vs eTLS in RCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_SFLvsOtherTLS_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.265, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.265 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange <= -0.265 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange >= 0.265 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in SFL_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Other_TLS'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
rccSFLagainstOtherTLS <- subset(Dat, expression =="Down")
rccOtherTLSaganistSFL <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 30),
    xlim = c(-2,2),
    pCutoff = 5*10e-3,
    FCcutoff = 0.265,
    pointSize = 2.0,
    labSize = 5.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 16,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_SFLvsOtherTLS_volcano_lfc0265_fdr005.pdf")
ggsave(file_name, width = 7, height = 10)







```

```{r DESeq2 results SFL vs Kidney}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype3_Kidney_vs_SFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_SFLTLSvsKidney_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)

```

```{r Volcano SFL vs Kidney in RCC}
##DESeq2 table
Dat  <- read.csv("DESeq2_rcc_SFLTLSvsKidney_LOQ3.csv")


Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -0.585 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 0.585 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in SFL_TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Kidney'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'  

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
#rccSFLTLSagainstKidney <- subset(Dat, expression =="Down")
#rccKidneyaganistSFLTLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 80),
    xlim = c(-5,4),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_rccLOQ3_SFLTLSvsKID_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 6, height = 6)

```

```{r DESeq2 normalisation SFL_TLS agains other TLS}
## set reference group for comparisons
dds$Tissue_Subtype <- relevel(dds$Tissue_Subtype, ref = "SFL_TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results SFL vs eTLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype_E_TLS_vs_SFL_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_rcc_SFLvsETLS_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

#List DEGs of maturation stages in RCC
```{r}
rccTCELLSagainstOtherTLS 
rccOtherTLSaganistTCELL

rccETLSagainstTCELL 
rccTCELLaganistETLS

rccETLSagainstPSFLTLS 
rccPSFTLSaganistETLS 

rccPFLagainstETCELL 
rccETCELLaganistPFL 

rccPFLTLSagainstSFLTLS 
rccSFTLSaganistPFLTLS 

rccSFLagainstOtherTLS 
rccOtherTLSaganistSFL 

TLSmaturationDEGsrcc <- sort(unique(c(rccTCELLSagainstOtherTLS$X, rccTCELLaganistETLS$X,rccOtherTLSaganistTCELL$X, rccETLSagainstTCELL$X, rccETLSagainstPSFLTLS$X, rccPSFTLSaganistETLS$X, rccPFLagainstETCELL$X,rccETCELLaganistPFL$X, rccPFLTLSagainstSFLTLS$X,rccSFTLSaganistPFLTLS$X, rccSFLagainstOtherTLS$X, rccOtherTLSaganistSFL$X)))

#AGAINST KIDNEY
rccTCELLSagainstKidney
rccETLSagainstKidney 
rccPFLTLSagainstKidney 
rccSFLTLSagainstKidney 

#rccKidneyaganistSFLTLS 
#rccKidneyaganistPFLTLS 
#rccKidneyaganistTCELL 
#rccKidneyaganistETLS

```

```{r total TLS DEGs against kidney}
TLSDEG_foldchange_Kidney <- merge(rccTCELLSagainstKidney[,c(1,3)],rccETLSagainstKidney[,c(1,3)], by = "X", all = TRUE)
TLSDEG_foldchange_Kidney <- merge(TLSDEG_foldchange_Kidney, rccPFLTLSagainstKidney[,c(1,3)], by = "X", all = TRUE)
TLSDEG_foldchange_Kidney <- merge(TLSDEG_foldchange_Kidney, rccSFLTLSagainstKidney[,c(1,3)], by = "X", all = TRUE)

colnames(TLSDEG_foldchange_Kidney) <- c("Gene", "Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS")
TLSDEG_foldchange_KidneyT <- as.data.frame(t(TLSDEG_foldchange_Kidney[,c(2:5)]))
colnames(TLSDEG_foldchange_KidneyT) <- TLSDEG_foldchange_Kidney$Gene
TLSDEG_foldchange_KidneyT <- (TLSDEG_foldchange_KidneyT *-1) 
TLSDEG_foldchange_KidneyT$TLStype <- as.factor(colnames(TLSDEG_foldchange_Kidney[,2:5]))

endcol <- ncol(TLSDEG_foldchange_KidneyT) -1

#setdiff(rccALLTLSagainstKIDNEY$X, TLSDEG_foldchange_Kidney$Gene)
```

```{r total TLS DEG fold change barplots}
dotplot_colour <- c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2"
                              )
    

i=2
for (i in 2:endcol) { 
pp_prefix <- "Barplot_DESeq2"
var <- "DEGfoldchange_inTLStypes"
 plot = colnames(TLSDEG_foldchange_KidneyT[i])
 
ggplot(data = TLSDEG_foldchange_KidneyT, aes(x = factor(TLStype, levels = c("Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS")), y = TLSDEG_foldchange_KidneyT[,i], fill = TLStype)) +
    geom_bar(stat="identity", position = 'identity') +
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5), limits=c(0.9, NA), oob=rescale_none, expand = expansion(mult = c(0.1, 0.1))) +
  #scale_color_aaas(values=dotplot_colour) +
   scale_fill_manual(values=dotplot_colour) +
    labs(x = "TLS type",
       y= bquote(~Log[2]~ 'FC versus kidney'),
      title = colnames(TLSDEG_foldchange_KidneyT[i])
      ) +
  #theme_pubr(border = TRUE) +
    theme_light() + 
    theme(text=element_text(size=22),
          plot.title = element_text(hjust = 0.5),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          axis.text.x = element_text(angle = 45, hjust = 1),
          legend.position = "none",
          strip.text = element_text(colour='black'))

fn <- sprintf("%s_%s_%s.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 4, height = 5)
}
```

```{r maturation TLS DEGs against kidney}
SFL  <- read.csv("DESeq2_rcc_SFLTLSvsKidney_LOQ3.csv")
PFL  <- read.csv("DESeq2_rcc_PFLTLSvsKidney_LOQ3.csv")
E  <- read.csv("DESeq2_rcc_ETLSvsKidney_LOQ3.csv")
TCELL  <- read.csv("DESeq2_rcc_TcellvsKidney_LOQ3.csv")

TLSmaturationDEGsrcc

TLSmaturDEGs_foldchnage_Kidney <- merge(TCELL[,c(1,3)],E[,c(1,3)], by = "X", all = TRUE)
TLSmaturDEGs_foldchnage_Kidney <- merge(TLSmaturDEGs_foldchnage_Kidney, PFL[,c(1,3)], by = "X", all = TRUE)
TLSmaturDEGs_foldchnage_Kidney <- merge(TLSmaturDEGs_foldchnage_Kidney, SFL[,c(1,3)], by = "X", all = TRUE)
colnames(TLSmaturDEGs_foldchnage_Kidney) <- c("Gene", "Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS")
TLSmaturDEGs_foldchnage_Kidney_sel <- TLSmaturDEGs_foldchnage_Kidney[TLSmaturDEGs_foldchnage_Kidney$Gene%in%TLSmaturationDEGsrcc,]

uniquelyTLSmaturationDEGsrcc <- setdiff(TLSmaturDEGs_foldchnage_Kidney_sel$Gene, TLSDEG_foldchange_Kidney$Gene)
TLSmaturDEGs_foldchnage_Kidney_sel2 <- TLSmaturDEGs_foldchnage_Kidney[TLSmaturDEGs_foldchnage_Kidney$Gene%in%uniquelyTLSmaturationDEGsrcc,]


TLSmaturDEGs_foldchnage_Kidney_sel2T <- as.data.frame(t(TLSmaturDEGs_foldchnage_Kidney_sel2[,c(2:5)]))
colnames(TLSmaturDEGs_foldchnage_Kidney_sel2T) <- TLSmaturDEGs_foldchnage_Kidney_sel2$Gene
TLSmaturDEGs_foldchnage_Kidney_sel2T <- (TLSmaturDEGs_foldchnage_Kidney_sel2T *-1) 
TLSmaturDEGs_foldchnage_Kidney_sel2T$TLStype <- as.factor(colnames(TLSmaturDEGs_foldchnage_Kidney_sel2[,2:5]))

endcol <- ncol(TLSmaturDEGs_foldchnage_Kidney_sel2T) -1

```
```{r maturation-unique TLS DEG fold change barplots}
dotplot_colour <- c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2"
                              )
    

i=2
for (i in 2:endcol) { 
pp_prefix <- "Barplot_DESeq2"
var <- "DEGfoldchange_inTLStypes"
 plot = colnames(TLSmaturDEGs_foldchnage_Kidney_sel2T[i])
 
ggplot(data = TLSmaturDEGs_foldchnage_Kidney_sel2T, aes(x = factor(TLStype, levels = c("Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS")), y = TLSmaturDEGs_foldchnage_Kidney_sel2T[,i], fill = TLStype)) +
    geom_bar(stat="identity", position = 'identity') +
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5), limits=c(NA, NA), expand = expansion(mult = c(0.1, 0.1))) +
  #scale_color_aaas(values=dotplot_colour) +
   scale_fill_manual(values=dotplot_colour) +
    labs(x = "TLS type",
       y= bquote(~Log[2]~ 'FC versus kidney'),
      title = colnames(TLSmaturDEGs_foldchnage_Kidney_sel2T[i])
      ) +
  #theme_pubr(border = TRUE) +
    theme_light() + 
    theme(text=element_text(size=22),
          plot.title = element_text(hjust = 0.5),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          axis.text.x = element_text(angle = 45, hjust = 1),
          legend.position = "none",
          strip.text = element_text(colour='black'))

fn <- sprintf("%s_%s_%s.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 4, height = 5)
}
```

#shared and unique DEGs in TLS maturation stages
```{r}
TLSmaturationDEGsrcc
TLSDEG_foldchange_Kidney$Gene

TLSmaturationDEGs
TLSDEG_foldchange_Lung$Gene

allDEGs <- sort(unique(c(TLSmaturationDEGsrcc, TLSDEG_foldchange_Kidney$Gene, TLSmaturationDEGs, TLSDEG_foldchange_Lung$Gene)))
```

#define TLS DEGs groups
```{r LSCC}
lscc_E_vs_T <- read.csv("DESeq2_lscc_ETLSvsTcellTLS_LOQ3.csv")
lscc_SFL_vs_E <- read.csv("DESeq2_lscc_SFLvsETLS_LOQ3.csv")
lscc_E_vs_Alv <- read.csv("DESeq2_lscc_ETLSvsAlveoles_LOQ3.csv")

LSCC_TLSDEG <- merge(lscc_E_vs_Alv[,c(1, 3, 5:6)], lscc_E_vs_T[,c(1, 3, 5:6)], by = "X", all = T)
LSCC_TLSDEG <- merge(LSCC_TLSDEG, lscc_SFL_vs_E[,c(1, 3, 5:6)], by = "X", all = T)
colnames(LSCC_TLSDEG) <- c("Gene", "E_Alv_lfc", "E_Alv_p", "E_Alv_padj", "E_Tc_lfc", "E_Tc_p", "E_Tc_padj", "SFL_E_lfc", "SFL_E_p", "SFL_E_padj")

LSCC_TLSDEG_sel <- LSCC_TLSDEG[LSCC_TLSDEG$Gene%in%allDEGs,]

LSCC_TLSDEG_sel$DEGgroup <- ifelse(LSCC_TLSDEG_sel$E_Alv_lfc < 0 & LSCC_TLSDEG_sel$E_Alv_p <0.05 & LSCC_TLSDEG_sel$E_Tc_p >= 0.05 & LSCC_TLSDEG_sel$SFL_E_p >= 0.05, "allTLS",
                            ifelse(LSCC_TLSDEG_sel$E_Alv_lfc < 0 & LSCC_TLSDEG_sel$E_Alv_p <0.05 & LSCC_TLSDEG_sel$E_Tc_p < 0.05 & LSCC_TLSDEG_sel$SFL_E_p >= 0.05 & LSCC_TLSDEG_sel$E_Tc_lfc < 0, "Bcell_tissueTLS",
                            ifelse(LSCC_TLSDEG_sel$E_Alv_lfc < 0 & LSCC_TLSDEG_sel$E_Alv_p <0.05 & LSCC_TLSDEG_sel$SFL_E_p < 0.05 & LSCC_TLSDEG_sel$SFL_E_lfc < 0, "SFL_tissueTLS",
                            ifelse(LSCC_TLSDEG_sel$E_Alv_lfc < 0 & LSCC_TLSDEG_sel$E_Alv_p <0.05 & LSCC_TLSDEG_sel$SFL_E_p < 0.05 & LSCC_TLSDEG_sel$SFL_E_lfc > 0 & LSCC_TLSDEG_sel$E_Tc_p >= 0.05, "E_tissueTLS",
                            ifelse(LSCC_TLSDEG_sel$E_Alv_p >= 0.05 & LSCC_TLSDEG_sel$SFL_E_p < 0.05 & LSCC_TLSDEG_sel$SFL_E_lfc < 0, "SFL_TLS",
                                   ifelse(LSCC_TLSDEG_sel$E_Alv_lfc > 0 & LSCC_TLSDEG_sel$E_Alv_p < 0.05 & LSCC_TLSDEG_sel$SFL_E_p < 0.05 & LSCC_TLSDEG_sel$SFL_E_lfc < 0, "SFL_TLS",
                            ifelse(LSCC_TLSDEG_sel$E_Alv_p >= 0.05 & LSCC_TLSDEG_sel$SFL_E_p < 0.05 & LSCC_TLSDEG_sel$SFL_E_lfc > 0, "E_TLS",
                                   ifelse(LSCC_TLSDEG_sel$E_Alv_lfc > 0 & LSCC_TLSDEG_sel$E_Alv_p < 0.05 & LSCC_TLSDEG_sel$SFL_E_p < 0.05 & LSCC_TLSDEG_sel$SFL_E_lfc > 0, "E_TLS",
                            ifelse(LSCC_TLSDEG_sel$E_Alv_lfc < 0 & LSCC_TLSDEG_sel$E_Alv_p < 0.05 & LSCC_TLSDEG_sel$E_Tc_p < 0.05 & LSCC_TLSDEG_sel$E_Tc_lfc > 0, "Tcell_tissueTLS",
                                   "Irrelevant")))))))))

table(LSCC_TLSDEG_sel$DEGgroup)
LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="Bcell_tissueTLS",]

```

```{r RCC}
rcc_E_vs_T <- read.csv("DESeq2_rcc_ETLSvsTcellTLS_LOQ3.csv")
rcc_SFL_vs_E <- read.csv("DESeq2_rcc_SFLvsETLS_LOQ3.csv")
rcc_E_vs_Kid <- read.csv("DESeq2_rcc_ETLSvsnormalKidney_LOQ3.csv")

RCC_TLSDEG <- merge(rcc_E_vs_Kid[,c(1, 3, 5:6)], rcc_E_vs_T[,c(1, 3, 5:6)], by = "X", all = T)
RCC_TLSDEG <- merge(RCC_TLSDEG, rcc_SFL_vs_E[,c(1, 3, 5:6)], by = "X", all = T)
colnames(RCC_TLSDEG) <- c("Gene", "E_Kid_lfc", "E_Kid_p", "E_Kid_padj", "E_Tc_lfc", "E_Tc_p", "E_Tc_padj", "SFL_E_lfc", "SFL_E_p", "SFL_E_padj")

RCC_TLSDEG_sel <- RCC_TLSDEG[RCC_TLSDEG$Gene%in%allDEGs,]


RCC_TLSDEG_sel$DEGgroup <- ifelse(RCC_TLSDEG_sel$E_Kid_lfc < 0 & RCC_TLSDEG_sel$E_Kid_p <0.05 & RCC_TLSDEG_sel$E_Tc_p >= 0.05 & RCC_TLSDEG_sel$SFL_E_p >= 0.05, "allTLS",
                                   ifelse(RCC_TLSDEG_sel$E_Kid_lfc < 0 & RCC_TLSDEG_sel$E_Kid_p <0.05 & RCC_TLSDEG_sel$E_Tc_p < 0.05 & RCC_TLSDEG_sel$SFL_E_p >= 0.05 & RCC_TLSDEG_sel$E_Tc_lfc < 0, "Bcell_tissueTLS",
                            ifelse(RCC_TLSDEG_sel$E_Kid_lfc < 0 & RCC_TLSDEG_sel$E_Kid_p <0.05 & RCC_TLSDEG_sel$SFL_E_p < 0.05 & RCC_TLSDEG_sel$SFL_E_lfc < 0, "SFL_tissueTLS",
                            ifelse(RCC_TLSDEG_sel$E_Kid_lfc < 0 & RCC_TLSDEG_sel$E_Kid_p <0.05 & RCC_TLSDEG_sel$SFL_E_p < 0.05 & RCC_TLSDEG_sel$SFL_E_lfc > 0 & RCC_TLSDEG_sel$E_Tc_p >= 0.05, "E_tissueTLS",
                            ifelse(RCC_TLSDEG_sel$E_Kid_p >= 0.05 & RCC_TLSDEG_sel$SFL_E_p < 0.05 & RCC_TLSDEG_sel$SFL_E_lfc < 0, "SFL_TLS",
                                   ifelse(RCC_TLSDEG_sel$E_Kid_lfc > 0 & RCC_TLSDEG_sel$E_Kid_p < 0.05 & RCC_TLSDEG_sel$SFL_E_p < 0.05 & RCC_TLSDEG_sel$SFL_E_lfc < 0, "SFL_TLS",
                            ifelse(RCC_TLSDEG_sel$E_Kid_p >= 0.05 & RCC_TLSDEG_sel$SFL_E_p < 0.05 & RCC_TLSDEG_sel$SFL_E_lfc > 0, "E_TLS",
                                   ifelse(RCC_TLSDEG_sel$E_Kid_lfc > 0 & RCC_TLSDEG_sel$E_Kid_p < 0.05 & RCC_TLSDEG_sel$SFL_E_p < 0.05 & RCC_TLSDEG_sel$SFL_E_lfc > 0, "E_TLS",
                            ifelse(RCC_TLSDEG_sel$E_Kid_lfc < 0 & RCC_TLSDEG_sel$E_Kid_p < 0.05 & RCC_TLSDEG_sel$E_Tc_p < 0.05 & RCC_TLSDEG_sel$E_Tc_lfc > 0, "Tcell_tissueTLS",
                                   "Irrelevant")))))))))
table(RCC_TLSDEG_sel$DEGgroup)
RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="Bcell_tissueTLS",]

```

```{r fuse both tumors}

#TLSDEGgrouping <- merge(LSCC_TLSDEG_sel[,c(1,11)],RCC_TLSDEG_sel[,c(1,11)], by= "Gene", all=T)
#colnames(TLSDEGgrouping) <- c("Gene", "LSCCgroup", "RCCgroup")
#write.csv(TLSDEGgrouping, "TLSDEG_grouping.csv")

#gooDEGs <- setdiff(allDEGs, commonIrrelevant)

LSCC_allTLS_unique <- setdiff(LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="allTLS",]$Gene, RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="allTLS",]$Gene)
RCC_allTLS_unique <- setdiff(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="allTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="allTLS",]$Gene)
commonAllTLSDEGs <- Reduce(intersect, list(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="allTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="allTLS",]$Gene))


LSCC_SFLtissueTLS_unique <- setdiff(LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="SFL_tissueTLS",]$Gene, RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="SFL_tissueTLS",]$Gene)
RCC_SFLtissueTLS_unique <- setdiff(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="SFL_tissueTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="SFL_tissueTLS",]$Gene)
commonSFLtissueTLSDEGs <- Reduce(intersect, list(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="SFL_tissueTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="SFL_tissueTLS",]$Gene))


LSCC_SFLTLS_unique <- setdiff(LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="SFL_TLS",]$Gene, RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="SFL_TLS",]$Gene)
RCC_SFLTLS_unique <- setdiff(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="SFL_tissueTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="SFL_TLS",]$Gene)
commonSFLTLSDEGs <- Reduce(intersect, list(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="SFL_TLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="SFL_TLS",]$Gene))


LSCC_BcTLS_unique <- setdiff(LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="Bcell_tissueTLS",]$Gene, RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="Bcell_tissueTLS",]$Gene)
RCC_BcTLS_unique <- setdiff(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="Bcell_tissueTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="Bcell_tissueTLS",]$Gene)
commonBcTLSDEGs <- Reduce(intersect, list(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="Bcell_tissueTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="Bcell_tissueTLS",]$Gene))


LSCC_TcTLS_unique <- setdiff(LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="Tcell_tissueTLS",]$Gene, RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="Tcell_tissueTLS",]$Gene)
RCC_TcTLS_unique <- setdiff(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="Tcell_tissueTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="Tcell_tissueTLS",]$Gene)
commonTcTLSDEGs <- Reduce(intersect, list(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="Tcell_tissueTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="Tcell_tissueTLS",]$Gene))


LSCC_ETLS_unique <- setdiff(LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="E_TLS",]$Gene, RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="E_TLS",]$Gene)
RCC_ETLS_unique <- setdiff(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="E_TLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="E_TLS",]$Gene)
commonETLSDEGs <- Reduce(intersect, list(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="E_TLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="E_TLS",]$Gene))


LSCC_EtissueTLS_unique <- setdiff(LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="E_tissueTLS",]$Gene, RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="E_tissueTLS",]$Gene)
RCC_EtissueTLS_unique <- setdiff(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="E_tissueTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="E_tissueTLS",]$Gene)
commonEtissueTLSDEGs <- Reduce(intersect, list(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="E_tissueTLS",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="E_tissueTLS",]$Gene))


LSCC_Irrelevant_unique <- setdiff(LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="Irrelevant",]$Gene, RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="Irrelevant",]$Gene)
RCC_Irrelevant_unique <- setdiff(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="Irrelevant",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="Irrelevant",]$Gene)
commonIrrelevant <- Reduce(intersect, list(RCC_TLSDEG_sel[RCC_TLSDEG_sel$DEGgroup=="Irrelevant",]$Gene, LSCC_TLSDEG_sel[LSCC_TLSDEG_sel$DEGgroup=="Irrelevant",]$Gene))




```

#Expression dot plots of all TLS DEGs
```{r load data}
cts <- as.matrix(m.tlstum.end)
Meta_data_reorganised <- pD.TLStum %>% mutate(Tissue_type = factor(Tissue_type))
unique(Meta_data_reorganised$Tissue_type)

Meta_data_reorganised <- Meta_data_reorganised %>% mutate(Tissue_type2 = dplyr::recode(Tissue_type,
  "Tum" = "Parenchyma",
  "Alv" = "Parenchyma",
  "Kid" = "Parenchyma"
  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_type2)

dim(cts)

table(dds$Tissue_type2)
```

```{r DESeq2 normalisation total TLS}
## set reference group for comparisons
dds$Tissue_type2 <- relevel(dds$Tissue_type2, ref = "TLS")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results total TLS}

resLFC <- lfcShrink(dds, coef="Tissue_type2_Parenchyma_vs_TLS", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
#write.csv(resOrdered, file="DESeq2_allTLSTUM_allTLSvsParenchyma_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

```{r Volcano total TLS vs Parench}
##DESeq2 table
Dat  <- read.csv("DESeq2_allTLSTUM_allTLSvsParenchyma_LOQ3.csv")

Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 2, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -1 ,'Down', 'Stable')
                     )

table(Dat$expression)

keyvals.colour <- ifelse(Dat$log2FoldChange < -1 & Dat$padj < 0.05, "#662965",
                  ifelse(Dat$log2FoldChange > 1 & Dat$padj < 0.05,"gold1", 
                      ifelse(Dat$padj >= 0.05,"#8F8F8F","steelblue4"
                      )))
                                
names(keyvals.colour)[keyvals.colour == "#662965"] <- 'Up in TLS'
names(keyvals.colour)[keyvals.colour == "gold1"] <- 'Up in Lung'
names(keyvals.colour)[keyvals.colour == "steelblue4"] <- 'Low fold change'
names(keyvals.colour)[keyvals.colour == '#8F8F8F'] <- 'NS'         

Df <- subset(Dat, expression !="Stable")
#for foldchange all change to >=0
ALLTLSagainstPARENCHYMA <- subset(Dat, expression =="Down")
PARENCHYMAaganistALLTLS <- subset(Dat, expression =="Up")

#significant <- as.vector(histounique$genes)
significant <- sort(as.vector(Df$X))

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 80),
    xlim = c(-3,4),
    pCutoff = 5*10e-3,
    FCcutoff = 1,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    colCustom = keyvals.colour,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_TLSvsLUNG_volcano_lfc1_fdr005.pdf")
ggsave(file_name, width = 8, height = 8)

```

```{r extract transformed values}
vsd <- vst(dds, blind=FALSE)
dim(assay(vsd))
DESeq2normexprALL <- t(assay(vsd))

#DEGs <- unique(c(TLSmaturationDEGsrcc, TLSDEG_foldchange_Kidney$Gene, TLSmaturationDEGs, TLSDEG_foldchange_Lung$Gene))
dataFC <- read.csv("AllDEGs_normalisedexpression_and_metadata.csv")
#colnames(dataFC)
allDEGs <- colnames(dataFC[25:594])
allDEGs <- gsub("\\.","-",allDEGs)

setdiff(allDEGs, colnames(DESeq2normexprALL))

#DESeq2normexpr <- as.data.frame(DESeq2normexprALL[,c(gooDEGs)])
DESeq2normexpr <- as.data.frame(DESeq2normexprALL[,c(allDEGs)])

unique(Meta_data_reorganised$Tissue_Subtype)
unique(Meta_data_reorganised$Tissue_type)
colnames(Meta_data_reorganised)
```

```{r set metadata}

data <- Meta_data_reorganised[,c(1:4, 23:30, 35:38, 40, 44,47,48)]
data <- data %>%
  mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
  "LSCC" = "Tum",
  "Alveoles" = "N",
  "SFL_TLS_B" = "SFL_TLS",
  "SFL_TLS_GC" = "SFL_TLS",
  "RCC" = "Tum",
  "Kidney" = "N"
  )) %>%
  mutate(Tissue_Subtypefacet = factor(Tissue_Subtype2, levels = c("N", "Tum", "Tcell_TLS", "E_TLS", "PFL_TLS", "SFL_TLS", "LN_T", "LN_B", "LN_GC"))) %>% 
  mutate(Tissue_type2 = dplyr::recode(Tissue_type,
  "Alv" = "N",
  "Kid" = "N",
  "Tum" = "T"
  )) %>% 
  mutate(Tissue_type2 = factor(Tissue_type2, levels = c("N", "T", "TLS", "LN")))

data1 <- cbind(data, DESeq2normexpr)

colnames(data1)
unique(data1$Tissue_Subtype)
endcol <-ncol(data1)

table(data1$Tumor_type, data1$Tissue_Subtypefacet)
#write.csv(data1, "AllDEGs_normalisedexpression_and_metadata.csv")

```

#Expression dot plots of all TLS DEGs in TLS subtypes
```{r colors}
pal <- c("#1F78C8", "#ff0000", "#33a02c", "#6A33C2", "#ff7f00", 
        "#565656", "#FFD700", "#a6cee3", "#FB6496", "#b2df8a", 
        "#CAB2D6", "#FDBF6F", "#999999", "#EEE685", "#C8308C", 
        "#FF83FA", "#C814FA", "#0000FF", "#36648B", "#00E2E5", 
        "#00FF00", "#778B00", "#BEBE00", "#8B3B00", "#A52A3C")

dotplot_colour =  c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2",
                              Tum = "#ff7f00",
                              N = "#FFD700",
                              LN_B = "#a6cee3",
                              LN_GC = "#FB6496",
                              LN_T = "#FF83FA"
                              )
    

```

```{r DEGs in AOI classes by subtype}


i=158
for (i in 24:endcol) {
 pp_prefix <- "Dotplot_DESeq2"
var <- "TLSTUMsubtypes"
 plot = colnames(data1[i])
ggplot(data = data1, aes(x = Tissue_Subtypefacet , y = data1[,i])) +
    geom_boxplot(alpha = 1, width = 0.6, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, width = 0.05, size=3, pch=21, aes(fill=Tissue_Subtypefacet), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
    #scale_y_continuous(trans = "log10") +
  scale_fill_manual(values=dotplot_colour) +
    facet_wrap(~Tumor_type, nrow = 2,) +
    stat_compare_means(comparisons=list(c("N", "E_TLS"), c("E_TLS", "SFL_TLS"), c("Tcell_TLS", "E_TLS")), size = 5) +
    stat_compare_means(comparisons=list(c("LN_T", "LN_B"), c("LN_B", "LN_GC"), c("SFL_TLS", "LN_GC")), size = 5) + 
  labs(x = "Tissue type",
       y= "Normalised expression",
       fill = "AOI subtype",
      title = colnames(data1[i])
      ) +
  #theme_pubr(border = TRUE) +
        theme_light() + 
    theme(text=element_text(size=20),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          strip.text = element_text(colour='black'),
        axis.text.x = element_text(angle = 55, hjust = 1, size=14),
          legend.title = element_text(size=14),
        legend.text = element_text(size=14), 
        legend.position = "none",
          plot.title = element_text(size=16, hjust = 0.5))

fn <- sprintf("%s_%s_%s.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 5, height = 6)



 }
```

```{r DEGs in AOI classes by subtype wo LNs}
data2 <- data1[data1$Tissue_type!="LN",]

i=29
for (i in 24:endcol) {
 pp_prefix <- "Dotplot_DESeq2"
var <- "TLSTUMsubtypes2"
 plot = colnames(data2[i])
ggplot(data = data2, aes(x = Tissue_Subtypefacet , y = data2[,i])) +
    geom_boxplot(alpha = 1, width = 0.6, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, width = 0.05, size=3, pch=21, aes(fill=Tissue_Subtypefacet), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
    #scale_y_continuous(trans = "log10") +
  scale_fill_manual(values=dotplot_colour) +
    facet_wrap(~Tumor_type, nrow = 2,) +
    stat_compare_means(comparisons=list(c("N", "E_TLS"), c("E_TLS", "SFL_TLS"), c("Tcell_TLS", "E_TLS")), size = 5) +
  labs(x = "Tissue type",
       y= "Normalised expression",
       fill = "AOI subtype",
      title = colnames(data2[i])
      ) +
  #theme_pubr(border = TRUE) +
        theme_light() + 
    theme(text=element_text(size=20),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          strip.text = element_text(colour='black'),
        axis.text.x = element_text(angle = 55, hjust = 1, size=14),
          legend.title = element_text(size=14),
        legend.text = element_text(size=14), 
        legend.position = "none",
          plot.title = element_text(size=16, hjust = 0.5))

fn <- sprintf("%s_%s_%s.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 3, height = 6)

 }
```

```{r DEGs in AOI classes by TLS subarea against LNs}
select <- c("Tcell_TLS", "SFL_TLS_B", "SFL_TLS_GC", "LN_T", "LN_B", "LN_GC")
data2 <- data1[data1$Tissue_Subtype%in%select,]
data2 <- data2[data2$Tumor_type=="LSCC",]%>%
  mutate(Tissue_Subtype = factor(Tissue_Subtype, levels = c("Tcell_TLS", "SFL_TLS_B", "SFL_TLS_GC", "LN_T", "LN_B", "LN_GC")))

i=29
for (i in 24:endcol) {
 pp_prefix <- "Dotplot_DESeq2"
var <- "TLSTUMsubtypes2"
 plot = colnames(data2[i])
ggplot(data = data2, aes(x = Tissue_Subtype , y = data2[,i])) +
    geom_boxplot(alpha = 1, width = 0.6, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, width = 0.05, size=3, pch=21, aes(fill=Tissue_Subtypefacet), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
    #scale_y_continuous(trans = "log10") +
  scale_fill_manual(values=dotplot_colour) +
    facet_wrap(~Tumor_type, nrow = 2,) +
    stat_compare_means(comparisons=list(c("SFL_TLS_GC", "LN_GC"), c("SFL_TLS_B", "SFL_TLS_GC"), c("LN_B", "LN_GC"), c("Tcell_TLS", "LN_T")), size = 5) +
  labs(x = "Tissue type",
       y= "Normalised expression",
       fill = "AOI subtype",
      title = colnames(data2[i])
      ) +
  #theme_pubr(border = TRUE) +
        theme_light() + 
    theme(text=element_text(size=20),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          strip.text = element_text(colour='black'),
        axis.text.x = element_text(angle = 55, hjust = 1, size=14),
          legend.title = element_text(size=14),
        legend.text = element_text(size=14), 
        legend.position = "none",
          plot.title = element_text(size=16, hjust = 0.5))

fn <- sprintf("%s_%s_%s_LN.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 3, height = 4)

 }
```


#Expression dot plots of total TLS DEGs in survival groups by subtype
```{r colors}
pal <- c("#1F78C8", "#ff0000", "#33a02c", "#6A33C2", "#ff7f00", 
        "#565656", "#FFD700", "#a6cee3", "#FB6496", "#b2df8a", 
        "#CAB2D6", "#FDBF6F", "#999999", "#EEE685", "#C8308C", 
        "#FF83FA", "#C814FA", "#0000FF", "#36648B", "#00E2E5", 
        "#00FF00", "#778B00", "#BEBE00", "#8B3B00", "#A52A3C")

dotplot_colour =  c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2",
                              Tum = "#ff7f00",
                              N = "#FFD700",
                              LN_B = "#a6cee3",
                              LN_GC = "#FB6496",
                              LN_T = "#FF83FA"
                              )
    

```

```{r DEGs in AOI classes by survival cat}
data2 <- data1[data1$Tissue_type!="LN",]

i=89
for (i in 24:endcol) {
 pp_prefix <- "Dotplot_DESeq2"
var <- "TLSTUMsurivalcats"
 plot = colnames(data2[i])
ggplot(data = data2, aes(x = Patient_Category , y = data2[,i])) +
    geom_boxplot(alpha = 1, width = 0.6, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, width = 0.05, size=3, pch=21, aes(fill=Tissue_Subtypefacet), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
    #scale_y_continuous(trans = "log10") +
  scale_fill_manual(values=dotplot_colour) +
    facet_grid(Tumor_type~Tissue_Subtypefacet) +
    stat_compare_means(comparisons=list(c("LTS", "STS")), size = 5) + 
  labs(x = "Patient category",
       y= "Normalised expression",
       fill = "AOI subtype",
      title = colnames(data2[i])
      ) +
  #theme_pubr(border = TRUE) +
        theme_light() + 
    theme(text=element_text(size=18),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          strip.text = element_text(colour='black'),
       # axis.text.x = element_text(angle = 55, hjust = 1, size=14),
          legend.title = element_text(size=14),
        legend.text = element_text(size=14), 
        legend.position = "none",
          plot.title = element_text(size=16, hjust = 0.5))

fn <- sprintf("%s_%s_%s.pdf", pp_prefix, var, plot)
 ggsave(filename = fn, width = 7, height = 5)



 }
```

#Expression dot plots of total TLS DEGs in tumors by subtype
```{r colors}
pal <- c("#1F78C8", "#ff0000", "#33a02c", "#6A33C2", "#ff7f00", 
        "#565656", "#FFD700", "#a6cee3", "#FB6496", "#b2df8a", 
        "#CAB2D6", "#FDBF6F", "#999999", "#EEE685", "#C8308C", 
        "#FF83FA", "#C814FA", "#0000FF", "#36648B", "#00E2E5", 
        "#00FF00", "#778B00", "#BEBE00", "#8B3B00", "#A52A3C")

dotplot_colour =  c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2",
                              Tum = "#ff7f00",
                              N = "#FFD700",
                              LN_B = "#a6cee3",
                              LN_GC = "#FB6496",
                              LN_T = "#FF83FA"
                              )
    

```

```{r DEGs in AOI classes by tumor}
data2 <- data1[data1$Tissue_type!="LN",]

i=151
for (i in 24:endcol) {
 pp_prefix <- "Dotplot_DESeq2"
var <- "TLSTUMtumortypes"
 plot = colnames(data2[i])
ggplot(data = data2, aes(x = Tumor_type , y = data2[,i])) +
    geom_boxplot(alpha = 1, width = 0.6, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, width = 0.05, size=3, pch=21, aes(fill=Tissue_Subtypefacet), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
    #scale_y_continuous(trans = "log10") +
  scale_fill_manual(values=dotplot_colour) +
    facet_wrap(~Tissue_Subtypefacet, nrow = 2,) +
    stat_compare_means(comparisons=list(c("LSCC", "RCC")), size = 5) + 
  labs(x = "Tissue type",
       y= "Normalised expression",
       fill = "AOI subtype",
      title = colnames(data2[i])
      ) +
  #theme_pubr(border = TRUE) +
        theme_light() + 
    theme(text=element_text(size=20),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          strip.text = element_text(colour='black'),
        axis.text.x = element_text(angle = 55, hjust = 1, size=14),
          legend.title = element_text(size=14),
        legend.text = element_text(size=14), 
        legend.position = "none",
          plot.title = element_text(size=16, hjust = 0.5))

fn <- sprintf("%s_%s_%s.pdf", pp_prefix, var,plot)
 ggsave(filename = fn, width = 5, height = 6)



 }
```


##FOLD CHNAGE clustering
```{r}
dataFC <- read.csv("AllDEGs_normalisedexpression_and_metadata.csv")
dataFClong <- reshape2::melt(dataFC,id = colnames(dataFC[,1:24]))
colnames(dataFClong) <- c(colnames(dataFC[,1:24]), "Gene", "NormExpr")

averageNT <- dataFClong[,c(8,11,25:26)] %>% group_by(Tumor_type, Tissue_Subtype, Gene) %>%
dplyr::summarize("average_expression" = mean(NormExpr))

#calculate fold change per gene per tumor type NormExpr/average_expr of Alveoles in LSCC AOIs and Kidney in RCC AOIs
averageNT_alveoles <- subset(averageNT, Tissue_Subtype=="Alveoles")
lung_expression = subset(dataFC,Tumor_type=="LSCC")[25:594] 
lung_meta <- subset(dataFC,Tumor_type=="LSCC")[1:24]
lung_expression_FC = data.frame(t(t(lung_expression)/averageNT_alveoles$average_expression))
lung_expression_FC <- cbind(lung_meta,lung_expression_FC)

averageNT_kidney <- subset(averageNT, Tissue_Subtype=="Kidney")
kidney_expression = subset(dataFC,Tumor_type=="RCC")[25:594] 
kidney_meta <- subset(dataFC,Tumor_type=="RCC")[1:24]
kidney_expression_FC = data.frame(t(t(kidney_expression)/averageNT_kidney$average_expression))
kidney_expression_FC <- cbind(kidney_meta, kidney_expression_FC)
```

#per AOI group FC pheatment
```{r}
averageNTwide <- dcast(averageNT, Tumor_type+Gene~Tissue_Subtype, value.var = "average_expression")

LSCC_E_FC <- averageNTwide[averageNTwide$Tumor_type=="LSCC",]$E_TLS/averageNTwide[averageNTwide$Tumor_type=="LSCC",]$Alveoles
LSCC_Tc_FC <- averageNTwide[averageNTwide$Tumor_type=="LSCC",]$Tcell_TLS/averageNTwide[averageNTwide$Tumor_type=="LSCC",]$Alveoles
LSCC_PFL_FC <- averageNTwide[averageNTwide$Tumor_type=="LSCC",]$PFL_TLS/averageNTwide[averageNTwide$Tumor_type=="LSCC",]$Alveoles
LSCC_SFL_FC <- averageNTwide[averageNTwide$Tumor_type=="LSCC",]$SFL_TLS/averageNTwide[averageNTwide$Tumor_type=="LSCC",]$Alveoles

LSCC_FC <- cbind(averageNTwide[averageNTwide$Tumor_type=="LSCC",1:2],LSCC_Tc_FC)
LSCC_FC <- cbind(LSCC_FC, LSCC_E_FC)
LSCC_FC <- cbind(LSCC_FC, LSCC_PFL_FC)
LSCC_FC <- cbind(LSCC_FC, LSCC_SFL_FC)

LSCC_FCmatrix <- LSCC_FC[,3:5]
rownames(LSCC_FCmatrix) <- LSCC_FC$Gene

```
```{r, heatmap all TLS vs lung, fig.width=6, fig.height=4}

hm <- as.matrix(LSCC_FCmatrix)

#colon annotations
annotation <- colnames(LSCC_FCmatrix)
 

my_colour = list(annotation = c(LSCC_SFL_FC = "#ff0000",
                              LSCC_E_FC = "#1F78C8",
                              LSCC_PFL_FC = "#33a02c",
                              LSCC_Tc_FC = "#6A33C2"
                              )
    )


breaklist <- seq(0, 2, by = 0.1) 
cols <- colorRampPalette(rev(brewer.pal(n=11, name = "RdYlBu")))(length(breaklist))


pdf("DESeq2_pheatmap_FC_totalTLSDEGs_LOQ3_pertissuesubtype_woPFL.pdf", width = 3, height = 20)
ph <- pheatmap::pheatmap(hm, color = cols, breaks = breaklist, cluster_cols = F, clustering_distance_rows = "euclidean", clustering_method = "ward.D2", cutree_rows = 10, fontsize = 3)
dev.off()
#clustering_distance_cols = "euclidean", clustering_distance_rows = "euclidean", clustering_method = "ward.D",

```

#per AOI FC pheatmap
```{r}
#pheatmap clustering of fold changes
expressionFC <- rbind(lung_expression_FC, kidney_expression_FC)
expressionFCsel <- expressionFC[expressionFC$Tissue_type2=="TLS",]
rownames(expressionFCsel) <- expressionFCsel$X
expressionFCsel <- expressionFCsel[order(expressionFCsel$Tissue_Subtype2, expressionFCsel$Tumor_type),]
FCmatrix <- t(expressionFCsel[,c(25:594)])

```

```{r, heatmap all TLS vs lung, fig.width=6, fig.height=4}

hm <- FCmatrix

#colon annotations
annotation <- expressionFCsel[,c("Tissue_Subtype2", "Tumor_type", "Patient_ID")]
table(annotation$Tumor_type)

my_colour = list(Tissue_Subtype2 = c(SFL_TLS = "#ff0000",
                              E_TLS = "#1F78C8",
                              PFL_TLS = "#33a02c",
                              Tcell_TLS = "#6A33C2"
                              ),
                 Tumor_type = c(RCC = "maroon",
                   LSCC = "darkturquoise"
                 ),
                #Patient_Category = c(STS = "yellow3",
                 #     LTS = "steelblue4"),
                Patient_ID = c("32288" = "#CAB2D6",
                              "17776" = "#FDBF6F",
                              "24136" = "maroon",
                              "16690" = "skyblue2",
                              "38234" = "deeppink1",
                              "39160" = "palegreen2",
                              "r19101" = "brown",
                              "r24784" = "khaki2",
                              "r15660" = "#6A3D9A",
                              "r8527" = "dodgerblue2",
                              "r30616" = "orchid1",
                              "r18773" = "green4"
                              )
    )


ph <- pheatmap::pheatmap(scale(hm), annotation_col = annotation, annotation_colors = my_colour)
range(scale(hm))

breaklist <- seq(0, 2, by = 0.1) 
cols <- colorRampPalette(rev(brewer.pal(n=11, name = "RdYlBu")))(length(breaklist))


pdf("DESeq2_pheatmap_FC_totalTLSDEGs_LOQ3scaled_ward2.pdf", width = 10, height = 10)
ph <- pheatmap::pheatmap(hm, color = cols, breaks = breaklist, annotation_col = annotation, annotation_colors = my_colour, cluster_cols = F, clustering_distance_cols = "euclidean", clustering_method = "ward.D", labels_col = "", cutree_cols = 5)
dev.off()
#clustering_distance_cols = "euclidean", clustering_distance_rows = "euclidean", clustering_method = "ward.D",

```







### TLS ep vs alveoles
```{r load data}
cts <- as.matrix(m.ep.end)
Meta_data_reorganised <- pD.ep %>% mutate(Tissue_Subtype = factor(Tissue_Subtype))
unique(Meta_data_reorganised$Tissue_Subtype)

#Meta_data_reorganised <- Meta_data_reorganised %>% mutate(Tissue_Subtype2 = dplyr::recode(Tissue_Subtype,
#  "LSCC" = "Lung",
 # "Alveoles" = "Lung",
 # "SFL_TLS_B" = "SFL_TLS",
 # "SFL_TLS_GC" = "SFL_TLS"
#  ))

coldata <- Meta_data_reorganised

#verify that the sample order matches in the matrix and metadata
all(rownames(coldata) == colnames(cts))

dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design= ~ Tissue_Subtype)

dim(cts)

table(dds$Tissue_Subtype)
```

```{r DESeq2 normalisation SFL}
## set reference group for comparisons
dds$Tissue_Subtype <- relevel(dds$Tissue_Subtype, ref = "Alveoles")

## normalise and compare groups
dds <- DESeq(dds)

## list the comparisons
resultsNames(dds)

```

```{r DESeq2 results EpithTLS}

resLFC <- lfcShrink(dds, coef="Tissue_Subtype_TLSep_vs_Alveoles", type="apeglm")
resLFC
summary(resLFC)
resOrdered <- as.data.frame(resLFC[order(resLFC$padj),])
write.csv(resOrdered, file="DESeq2_TLSepvsALV_LOQ3.csv")
resOrdered$Gene <- rownames(resOrdered)


```

#Volcanos versus Lung
```{r Volcano Epith vs Lung}
##DESeq2 table
Dat  <- read.csv("DESeq2_TLSepvsALV_LOQ3.csv")

colnames(Dat)

Dat$expression = ifelse(Dat$padj < 0.05 & Dat$log2FoldChange >= 0.585, 'Up',
                     ifelse(Dat$padj < 0.05 & Dat$log2FoldChange <= -0.585 ,'Down', 'Stable')
                     )

table(Dat$expression)

Df <- subset(Dat, expression !="Stable")
View(Df)
#for foldchange all change to >=0
TLSepithagainstALV <- subset(Dat, expression =="Up")
ALVaganistTLSepith <- subset(Dat, expression =="Down")

#significant <- as.vector(histounique$genes)
significant <- as.vector(Df$X)

options(ggrepel.max.overlaps = Inf)

EnhancedVolcano(Dat,
    lab = Dat$X,
    x = 'log2FoldChange',
    y = 'padj',
    selectLab = c(significant),
    xlab = bquote(~Log[2]~ 'fold change'),
    ylab = bquote("-"~Log[10]~ 'adj. p-val'),
    ylim = c(0, 8),
    xlim = c(-3,3),
    pCutoff = 5*10e-3,
    FCcutoff = 0.585,
    pointSize = 2.0,
    labSize = 4.0,
    maxoverlapsConnectors = Inf,
    #shape = c(4, 35, 17, 18),
    colAlpha = 1,
    legendPosition = "top" ,
    legendLabSize = 14,
    legendIconSize = 5.0,
    drawConnectors = TRUE,
    widthConnectors = 0.25)
 # +    scale_y_continuous(trans = log2_trans())


file_name <- paste("DESeq2_lsccLOQ3_TLSEpithvsAlv_volcano_lfc0585_fdr005.pdf")
ggsave(file_name, width = 8, height = 8)

```

```{r extract transformed values}
vsd <- vst(dds, blind=FALSE)
#rld <- rlog(dds, blind=FALSE)

dim(assay(vsd))
DESeq2normexprALL <- t(assay(vsd))
DESeq2normexpr <- as.data.frame(DESeq2normexprALL[,TLSepithagainstALV$X])

colnames(Meta_data_reorganised)
data <- Meta_data_reorganised[,c(1:4, 23:30, 35:38, 40, 44,47,48)]

#data <- cbind(data, DESeq2normexpr)

colnames(data)
unique(data$Patient_Category)


data1 <- cbind(data,DESeq2normexpr)
colnames(data1)
endcol <-ncol(data1)
```

#between tissue types
```{r DEGs in ALV vs EPITH }

i=21
for (i in 21:endcol) {

ggplot(data = data1, aes(x = Tissue_Subtype, y = data1[,i])) +
    geom_boxplot(alpha = 1, width = 0.4, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, width = 0.05, size=3, pch=21, aes(fill=Patient_Category), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
  #scale_color_aaas() +
  scale_fill_aaas() +
    #facet_wrap(~Tissue_Subtypefacet, nrow = 1) +
    stat_compare_means(comparisons=list(c("Alveoles", "TLSep")), size = 5) +
  labs(x = "Tissue type",
       y= "Normalised expression", #expression(paste("GC/cm"^2)),
      title = colnames(data1[i])
      ) +
  #theme_pubr(border = TRUE) +
    theme_light()+
    #theme(legend.position="none")+
  theme(text=element_text(size=12),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          #strip.text = element_text(color='grey'),
        #axis.text.x = element_text(angle = 45, hjust = 1),
          legend.title = element_text(size=12),
          legend.text = element_text(size=12), 
          plot.title = element_text(size=14, hjust = 0.5))
  

ggsave(filename = paste("DESeq2norm", colnames(data1[i]), "inTissuetypes_LOQ2.pdf", sep="_"), width = 6, height = 6)

 }
```

```{r DEGs in ALV vs EPITH facet}

i=21
for (i in 21:endcol) {

ggplot(data = data1, aes(x = Tissue_Subtype, y = data1[,i])) +
    geom_boxplot(alpha = 1, width = 0.4, outlier.shape = NA, lwd = 0.5) +
    geom_point(alpha = 1, width = 0.05, size=3, pch=21, aes(fill=Patient_Category), position = position_jitter(width = 0.2, height = 0), show.legend = TRUE) + #aes(fill=Patient_Category),
    scale_y_continuous(breaks = scales::pretty_breaks(n = 5),limits = c(NA,NA), expand = expansion(mult = c(0.1, 0.1))) +
  #scale_color_aaas() +
  scale_fill_aaas() +
    facet_wrap(~Patient_ID, nrow = 1) +
    stat_compare_means(comparisons=list(c("Alveoles", "TLSep")), size = 5) +
  labs(x = "Tissue type",
       y= "Normalised expression", #expression(paste("GC/cm"^2)),
      title = colnames(data1[i])
      ) +
  #theme_pubr(border = TRUE) +
    theme_light()+
    #theme(legend.position="none")+
  theme(text=element_text(size=12),
          panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          #strip.text = element_text(color='grey'),
        axis.text.x = element_text(angle = 45, hjust = 1),
          legend.title = element_text(size=12),
          legend.text = element_text(size=12), 
          plot.title = element_text(size=14, hjust = 0.5))
  

ggsave(filename = paste("DESeq2norm", colnames(data1[i]), "inTissuetypes_LOQ2patientfacet.pdf", sep="_"), width = 8, height = 6)

 }
```
